<!doctype html>
<html>
<head>
<meta charset='UTF-8'><meta name='viewport' content='width=device-width initial-scale=1'>

<link href='file:///home/ninomoriaty/.config/Typora/themes/blubook/fonts.css' rel='stylesheet' type='text/css' /><style type='text/css'>html {overflow-x: initial !important;}:root { --bg-color:#ffffff; --text-color:#333333; --select-text-bg-color:#B5D6FC; --select-text-font-color:auto; --monospace:"Lucida Console",Consolas,"Courier",monospace; --title-bar-height:20px; }
.mac-os-11 { --title-bar-height:28px; }
html { font-size: 14px; background-color: var(--bg-color); color: var(--text-color); font-family: "Helvetica Neue", Helvetica, Arial, sans-serif; -webkit-font-smoothing: antialiased; }
body { margin: 0px; padding: 0px; height: auto; bottom: 0px; top: 0px; left: 0px; right: 0px; font-size: 1rem; line-height: 1.42857; overflow-x: hidden; background: inherit; tab-size: 4; }
iframe { margin: auto; }
a.url { word-break: break-all; }
a:active, a:hover { outline: 0px; }
.in-text-selection, ::selection { text-shadow: none; background: var(--select-text-bg-color); color: var(--select-text-font-color); }
#write { margin: 0px auto; height: auto; width: inherit; word-break: normal; overflow-wrap: break-word; position: relative; white-space: normal; overflow-x: visible; padding-top: 36px; }
#write.first-line-indent p { text-indent: 2em; }
#write.first-line-indent li p, #write.first-line-indent p * { text-indent: 0px; }
#write.first-line-indent li { margin-left: 2em; }
.for-image #write { padding-left: 8px; padding-right: 8px; }
body.typora-export { padding-left: 30px; padding-right: 30px; }
.typora-export .footnote-line, .typora-export li, .typora-export p { white-space: pre-wrap; }
.typora-export .task-list-item input { pointer-events: none; }
@media screen and (max-width: 500px) {
  body.typora-export { padding-left: 0px; padding-right: 0px; }
  #write { padding-left: 20px; padding-right: 20px; }
  .CodeMirror-sizer { margin-left: 0px !important; }
  .CodeMirror-gutters { display: none !important; }
}
#write li > figure:last-child { margin-bottom: 0.5rem; }
#write ol, #write ul { position: relative; }
img { max-width: 100%; vertical-align: middle; image-orientation: from-image; }
button, input, select, textarea { color: inherit; font: inherit; }
input[type="checkbox"], input[type="radio"] { line-height: normal; padding: 0px; }
*, ::after, ::before { box-sizing: border-box; }
#write h1, #write h2, #write h3, #write h4, #write h5, #write h6, #write p, #write pre { width: inherit; }
#write h1, #write h2, #write h3, #write h4, #write h5, #write h6, #write p { position: relative; }
p { line-height: inherit; }
h1, h2, h3, h4, h5, h6 { break-after: avoid-page; break-inside: avoid; orphans: 4; }
p { orphans: 4; }
h1 { font-size: 2rem; }
h2 { font-size: 1.8rem; }
h3 { font-size: 1.6rem; }
h4 { font-size: 1.4rem; }
h5 { font-size: 1.2rem; }
h6 { font-size: 1rem; }
.md-math-block, .md-rawblock, h1, h2, h3, h4, h5, h6, p { margin-top: 1rem; margin-bottom: 1rem; }
.hidden { display: none; }
.md-blockmeta { color: rgb(204, 204, 204); font-weight: 700; font-style: italic; }
a { cursor: pointer; }
sup.md-footnote { padding: 2px 4px; background-color: rgba(238, 238, 238, 0.7); color: rgb(85, 85, 85); border-radius: 4px; cursor: pointer; }
sup.md-footnote a, sup.md-footnote a:hover { color: inherit; text-transform: inherit; text-decoration: inherit; }
#write input[type="checkbox"] { cursor: pointer; width: inherit; height: inherit; }
figure { overflow-x: auto; margin: 1.2em 0px; max-width: calc(100% + 16px); padding: 0px; }
figure > table { margin: 0px; }
tr { break-inside: avoid; break-after: auto; }
thead { display: table-header-group; }
table { border-collapse: collapse; border-spacing: 0px; width: 100%; overflow: auto; break-inside: auto; text-align: left; }
table.md-table td { min-width: 32px; }
.CodeMirror-gutters { border-right: 0px; background-color: inherit; }
.CodeMirror-linenumber { user-select: none; }
.CodeMirror { text-align: left; }
.CodeMirror-placeholder { opacity: 0.3; }
.CodeMirror pre { padding: 0px 4px; }
.CodeMirror-lines { padding: 0px; }
div.hr:focus { cursor: none; }
#write pre { white-space: pre-wrap; }
#write.fences-no-line-wrapping pre { white-space: pre; }
#write pre.ty-contain-cm { white-space: normal; }
.CodeMirror-gutters { margin-right: 4px; }
.md-fences { font-size: 0.9rem; display: block; break-inside: avoid; text-align: left; overflow: visible; white-space: pre; background: inherit; position: relative !important; }
.md-fences-adv-panel { width: 100%; margin-top: 10px; text-align: center; padding-top: 0px; padding-bottom: 8px; overflow-x: auto; }
#write .md-fences.mock-cm { white-space: pre-wrap; }
.md-fences.md-fences-with-lineno { padding-left: 0px; }
#write.fences-no-line-wrapping .md-fences.mock-cm { white-space: pre; overflow-x: auto; }
.md-fences.mock-cm.md-fences-with-lineno { padding-left: 8px; }
.CodeMirror-line, twitterwidget { break-inside: avoid; }
.footnotes { opacity: 0.8; font-size: 0.9rem; margin-top: 1em; margin-bottom: 1em; }
.footnotes + .footnotes { margin-top: 0px; }
.md-reset { margin: 0px; padding: 0px; border: 0px; outline: 0px; vertical-align: top; background: 0px 0px; text-decoration: none; text-shadow: none; float: none; position: static; width: auto; height: auto; white-space: nowrap; cursor: inherit; -webkit-tap-highlight-color: transparent; line-height: normal; font-weight: 400; text-align: left; box-sizing: content-box; direction: ltr; }
li div { padding-top: 0px; }
blockquote { margin: 1rem 0px; }
li .mathjax-block, li p { margin: 0.5rem 0px; }
li blockquote { margin: 1rem 0px; }
li { margin: 0px; position: relative; }
blockquote > :last-child { margin-bottom: 0px; }
blockquote > :first-child, li > :first-child { margin-top: 0px; }
.footnotes-area { color: rgb(136, 136, 136); margin-top: 0.714rem; padding-bottom: 0.143rem; white-space: normal; }
#write .footnote-line { white-space: pre-wrap; }
@media print {
  body, html { border: 1px solid transparent; height: 99%; break-after: avoid; break-before: avoid; font-variant-ligatures: no-common-ligatures; }
  #write { margin-top: 0px; padding-top: 0px; border-color: transparent !important; }
  .typora-export * { -webkit-print-color-adjust: exact; }
  .typora-export #write { break-after: avoid; }
  .typora-export #write::after { height: 0px; }
  .is-mac table { break-inside: avoid; }
  .typora-export-show-outline .typora-export-sidebar { display: none; }
}
.footnote-line { margin-top: 0.714em; font-size: 0.7em; }
a img, img a { cursor: pointer; }
pre.md-meta-block { font-size: 0.8rem; min-height: 0.8rem; white-space: pre-wrap; background: rgb(204, 204, 204); display: block; overflow-x: hidden; }
p > .md-image:only-child:not(.md-img-error) img, p > img:only-child { display: block; margin: auto; }
#write.first-line-indent p > .md-image:only-child:not(.md-img-error) img { left: -2em; position: relative; }
p > .md-image:only-child { display: inline-block; width: 100%; }
#write .MathJax_Display { margin: 0.8em 0px 0px; }
.md-math-block { width: 100%; }
.md-math-block:not(:empty)::after { display: none; }
.MathJax_ref { fill: currentcolor; }
[contenteditable="true"]:active, [contenteditable="true"]:focus, [contenteditable="false"]:active, [contenteditable="false"]:focus { outline: 0px; box-shadow: none; }
.md-task-list-item { position: relative; list-style-type: none; }
.task-list-item.md-task-list-item { padding-left: 0px; }
.md-task-list-item > input { position: absolute; top: 0px; left: 0px; margin-left: -1.2em; margin-top: calc(1em - 10px); border: none; }
.math { font-size: 1rem; }
.md-toc { min-height: 3.58rem; position: relative; font-size: 0.9rem; border-radius: 10px; }
.md-toc-content { position: relative; margin-left: 0px; }
.md-toc-content::after, .md-toc::after { display: none; }
.md-toc-item { display: block; color: rgb(65, 131, 196); }
.md-toc-item a { text-decoration: none; }
.md-toc-inner:hover { text-decoration: underline; }
.md-toc-inner { display: inline-block; cursor: pointer; }
.md-toc-h1 .md-toc-inner { margin-left: 0px; font-weight: 700; }
.md-toc-h2 .md-toc-inner { margin-left: 2em; }
.md-toc-h3 .md-toc-inner { margin-left: 4em; }
.md-toc-h4 .md-toc-inner { margin-left: 6em; }
.md-toc-h5 .md-toc-inner { margin-left: 8em; }
.md-toc-h6 .md-toc-inner { margin-left: 10em; }
@media screen and (max-width: 48em) {
  .md-toc-h3 .md-toc-inner { margin-left: 3.5em; }
  .md-toc-h4 .md-toc-inner { margin-left: 5em; }
  .md-toc-h5 .md-toc-inner { margin-left: 6.5em; }
  .md-toc-h6 .md-toc-inner { margin-left: 8em; }
}
a.md-toc-inner { font-size: inherit; font-style: inherit; font-weight: inherit; line-height: inherit; }
.footnote-line a:not(.reversefootnote) { color: inherit; }
.md-attr { display: none; }
.md-fn-count::after { content: "."; }
code, pre, samp, tt { font-family: var(--monospace); }
kbd { margin: 0px 0.1em; padding: 0.1em 0.6em; font-size: 0.8em; color: rgb(36, 39, 41); background: rgb(255, 255, 255); border: 1px solid rgb(173, 179, 185); border-radius: 3px; box-shadow: rgba(12, 13, 14, 0.2) 0px 1px 0px, rgb(255, 255, 255) 0px 0px 0px 2px inset; white-space: nowrap; vertical-align: middle; }
.md-comment { color: rgb(162, 127, 3); opacity: 0.8; font-family: var(--monospace); }
code { text-align: left; vertical-align: initial; }
a.md-print-anchor { white-space: pre !important; border-width: initial !important; border-style: none !important; border-color: initial !important; display: inline-block !important; position: absolute !important; width: 1px !important; right: 0px !important; outline: 0px !important; background: 0px 0px !important; text-decoration: initial !important; text-shadow: initial !important; }
.md-inline-math .MathJax_SVG .noError { display: none !important; }
.html-for-mac .inline-math-svg .MathJax_SVG { vertical-align: 0.2px; }
.md-fences-math .MathJax_SVG_Display, .md-math-block .MathJax_SVG_Display { text-align: center; margin: 0px; position: relative; text-indent: 0px; max-width: none; max-height: none; min-height: 0px; min-width: 100%; width: auto; overflow-y: visible; display: block !important; }
.MathJax_SVG_Display, .md-inline-math .MathJax_SVG_Display { width: auto; margin: inherit; display: inline-block !important; }
.MathJax_SVG .MJX-monospace { font-family: var(--monospace); }
.MathJax_SVG .MJX-sans-serif { font-family: sans-serif; }
.MathJax_SVG { display: inline; font-style: normal; font-weight: 400; line-height: normal; text-indent: 0px; text-align: left; text-transform: none; letter-spacing: normal; word-spacing: normal; overflow-wrap: normal; white-space: nowrap; float: none; direction: ltr; max-width: none; max-height: none; min-width: 0px; min-height: 0px; border: 0px; padding: 0px; margin: 0px; zoom: 90%; }
#math-inline-preview-content { zoom: 1.1; }
.MathJax_SVG * { transition: none 0s ease 0s; }
.MathJax_SVG_Display svg { vertical-align: middle !important; margin-bottom: 0px !important; margin-top: 0px !important; }
.os-windows.monocolor-emoji .md-emoji { font-family: "Segoe UI Symbol", sans-serif; }
.md-diagram-panel > svg { max-width: 100%; }
[lang="flow"] svg, [lang="mermaid"] svg { max-width: 100%; height: auto; }
[lang="mermaid"] .node text { font-size: 1rem; }
table tr th { border-bottom: 0px; }
video { max-width: 100%; display: block; margin: 0px auto; }
iframe { max-width: 100%; width: 100%; border: none; }
.highlight td, .highlight tr { border: 0px; }
mark { background: rgb(255, 255, 0); color: rgb(0, 0, 0); }
.md-html-inline .md-plain, .md-html-inline strong, mark .md-inline-math, mark strong { color: inherit; }
.md-expand mark .md-meta { opacity: 0.3 !important; }
mark .md-meta { color: rgb(0, 0, 0); }
@media print {
  .typora-export h1, .typora-export h2, .typora-export h3, .typora-export h4, .typora-export h5, .typora-export h6 { break-inside: avoid; }
}
.md-diagram-panel .messageText { stroke: none !important; }
.md-diagram-panel .start-state { fill: var(--node-fill); }
.md-diagram-panel .edgeLabel rect { opacity: 1 !important; }
.md-require-zoom-fix foreignobject { font-size: var(--mermaid-font-zoom); }
.md-fences.md-fences-math { font-size: 1em; }
.md-fences-math .MathJax_SVG_Display { margin-top: 8px; cursor: default; }
.md-fences-advanced:not(.md-focus) { padding: 0px; white-space: nowrap; border: 0px; }
.md-fences-advanced:not(.md-focus) { background: inherit; }
.typora-export-show-outline .typora-export-content { max-width: 1440px; margin: auto; display: flex; flex-direction: row; }
.typora-export-sidebar { width: 300px; font-size: 0.8rem; margin-top: 80px; margin-right: 18px; }
.typora-export-show-outline #write { --webkit-flex:2; flex: 2 1 0%; }
.typora-export-sidebar .outline-content { position: fixed; top: 0px; max-height: 100%; overflow: hidden auto; padding-bottom: 30px; padding-top: 60px; width: 300px; }
@media screen and (max-width: 1024px) {
  .typora-export-sidebar, .typora-export-sidebar .outline-content { width: 240px; }
}
@media screen and (max-width: 800px) {
  .typora-export-sidebar { display: none; }
}
.outline-content li, .outline-content ul { margin-left: 0px; margin-right: 0px; padding-left: 0px; padding-right: 0px; list-style: none; }
.outline-content ul { margin-top: 0px; margin-bottom: 0px; }
.outline-content strong { font-weight: 400; }
.outline-expander { width: 1rem; height: 1.42857rem; position: relative; display: table-cell; vertical-align: middle; cursor: pointer; padding-left: 4px; }
.outline-expander::before { content: ""; position: relative; font-family: Ionicons; display: inline-block; font-size: 8px; vertical-align: middle; }
.outline-item { padding-top: 3px; padding-bottom: 3px; cursor: pointer; }
.outline-expander:hover::before { content: ""; }
.outline-h1 > .outline-item { padding-left: 0px; }
.outline-h2 > .outline-item { padding-left: 1em; }
.outline-h3 > .outline-item { padding-left: 2em; }
.outline-h4 > .outline-item { padding-left: 3em; }
.outline-h5 > .outline-item { padding-left: 4em; }
.outline-h6 > .outline-item { padding-left: 5em; }
.outline-label { cursor: pointer; display: table-cell; vertical-align: middle; text-decoration: none; color: inherit; }
.outline-label:hover { text-decoration: underline; }
.outline-item:hover { border-color: rgb(245, 245, 245); background-color: var(--item-hover-bg-color); }
.outline-item:hover { margin-left: -28px; margin-right: -28px; border-left: 28px solid transparent; border-right: 28px solid transparent; }
.outline-item-single .outline-expander::before, .outline-item-single .outline-expander:hover::before { display: none; }
.outline-item-open > .outline-item > .outline-expander::before { content: ""; }
.outline-children { display: none; }
.info-panel-tab-wrapper { display: none; }
.outline-item-open > .outline-children { display: block; }
.typora-export .outline-item { padding-top: 1px; padding-bottom: 1px; }
.typora-export .outline-item:hover { margin-right: -8px; border-right: 8px solid transparent; }
.typora-export .outline-expander::before { content: "+"; font-family: inherit; top: -1px; }
.typora-export .outline-expander:hover::before, .typora-export .outline-item-open > .outline-item > .outline-expander::before { content: "−"; }
.typora-export-collapse-outline .outline-children { display: none; }
.typora-export-collapse-outline .outline-item-open > .outline-children, .typora-export-no-collapse-outline .outline-children { display: block; }
.typora-export-no-collapse-outline .outline-expander::before { content: "" !important; }
.typora-export-show-outline .outline-item-active > .outline-item .outline-label { font-weight: 700; }


.CodeMirror { height: auto; }
.CodeMirror.cm-s-inner { background: inherit; }
.CodeMirror-scroll { overflow: auto hidden; z-index: 3; }
.CodeMirror-gutter-filler, .CodeMirror-scrollbar-filler { background-color: rgb(255, 255, 255); }
.CodeMirror-gutters { border-right: 1px solid rgb(221, 221, 221); background: inherit; white-space: nowrap; }
.CodeMirror-linenumber { padding: 0px 3px 0px 5px; text-align: right; color: rgb(153, 153, 153); }
.cm-s-inner .cm-keyword { color: rgb(119, 0, 136); }
.cm-s-inner .cm-atom, .cm-s-inner.cm-atom { color: rgb(34, 17, 153); }
.cm-s-inner .cm-number { color: rgb(17, 102, 68); }
.cm-s-inner .cm-def { color: rgb(0, 0, 255); }
.cm-s-inner .cm-variable { color: rgb(0, 0, 0); }
.cm-s-inner .cm-variable-2 { color: rgb(0, 85, 170); }
.cm-s-inner .cm-variable-3 { color: rgb(0, 136, 85); }
.cm-s-inner .cm-string { color: rgb(170, 17, 17); }
.cm-s-inner .cm-property { color: rgb(0, 0, 0); }
.cm-s-inner .cm-operator { color: rgb(152, 26, 26); }
.cm-s-inner .cm-comment, .cm-s-inner.cm-comment { color: rgb(170, 85, 0); }
.cm-s-inner .cm-string-2 { color: rgb(255, 85, 0); }
.cm-s-inner .cm-meta { color: rgb(85, 85, 85); }
.cm-s-inner .cm-qualifier { color: rgb(85, 85, 85); }
.cm-s-inner .cm-builtin { color: rgb(51, 0, 170); }
.cm-s-inner .cm-bracket { color: rgb(153, 153, 119); }
.cm-s-inner .cm-tag { color: rgb(17, 119, 0); }
.cm-s-inner .cm-attribute { color: rgb(0, 0, 204); }
.cm-s-inner .cm-header, .cm-s-inner.cm-header { color: rgb(0, 0, 255); }
.cm-s-inner .cm-quote, .cm-s-inner.cm-quote { color: rgb(0, 153, 0); }
.cm-s-inner .cm-hr, .cm-s-inner.cm-hr { color: rgb(153, 153, 153); }
.cm-s-inner .cm-link, .cm-s-inner.cm-link { color: rgb(0, 0, 204); }
.cm-negative { color: rgb(221, 68, 68); }
.cm-positive { color: rgb(34, 153, 34); }
.cm-header, .cm-strong { font-weight: 700; }
.cm-del { text-decoration: line-through; }
.cm-em { font-style: italic; }
.cm-link { text-decoration: underline; }
.cm-error { color: red; }
.cm-invalidchar { color: red; }
.cm-constant { color: rgb(38, 139, 210); }
.cm-defined { color: rgb(181, 137, 0); }
div.CodeMirror span.CodeMirror-matchingbracket { color: rgb(0, 255, 0); }
div.CodeMirror span.CodeMirror-nonmatchingbracket { color: rgb(255, 34, 34); }
.cm-s-inner .CodeMirror-activeline-background { background: inherit; }
.CodeMirror { position: relative; overflow: hidden; }
.CodeMirror-scroll { height: 100%; outline: 0px; position: relative; box-sizing: content-box; background: inherit; }
.CodeMirror-sizer { position: relative; }
.CodeMirror-gutter-filler, .CodeMirror-hscrollbar, .CodeMirror-scrollbar-filler, .CodeMirror-vscrollbar { position: absolute; z-index: 6; display: none; }
.CodeMirror-vscrollbar { right: 0px; top: 0px; overflow: hidden; }
.CodeMirror-hscrollbar { bottom: 0px; left: 0px; overflow: hidden; }
.CodeMirror-scrollbar-filler { right: 0px; bottom: 0px; }
.CodeMirror-gutter-filler { left: 0px; bottom: 0px; }
.CodeMirror-gutters { position: absolute; left: 0px; top: 0px; padding-bottom: 30px; z-index: 3; }
.CodeMirror-gutter { white-space: normal; height: 100%; box-sizing: content-box; padding-bottom: 30px; margin-bottom: -32px; display: inline-block; }
.CodeMirror-gutter-wrapper { position: absolute; z-index: 4; background: 0px 0px !important; border: none !important; }
.CodeMirror-gutter-background { position: absolute; top: 0px; bottom: 0px; z-index: 4; }
.CodeMirror-gutter-elt { position: absolute; cursor: default; z-index: 4; }
.CodeMirror-lines { cursor: text; }
.CodeMirror pre { border-radius: 0px; border-width: 0px; background: 0px 0px; font-family: inherit; font-size: inherit; margin: 0px; white-space: pre; overflow-wrap: normal; color: inherit; z-index: 2; position: relative; overflow: visible; }
.CodeMirror-wrap pre { overflow-wrap: break-word; white-space: pre-wrap; word-break: normal; }
.CodeMirror-code pre { border-right: 30px solid transparent; width: fit-content; }
.CodeMirror-wrap .CodeMirror-code pre { border-right: none; width: auto; }
.CodeMirror-linebackground { position: absolute; left: 0px; right: 0px; top: 0px; bottom: 0px; z-index: 0; }
.CodeMirror-linewidget { position: relative; z-index: 2; overflow: auto; }
.CodeMirror-wrap .CodeMirror-scroll { overflow-x: hidden; }
.CodeMirror-measure { position: absolute; width: 100%; height: 0px; overflow: hidden; visibility: hidden; }
.CodeMirror-measure pre { position: static; }
.CodeMirror div.CodeMirror-cursor { position: absolute; visibility: hidden; border-right: none; width: 0px; }
.CodeMirror div.CodeMirror-cursor { visibility: hidden; }
.CodeMirror-focused div.CodeMirror-cursor { visibility: inherit; }
.cm-searching { background: rgba(255, 255, 0, 0.4); }
span.cm-underlined { text-decoration: underline; }
span.cm-strikethrough { text-decoration: line-through; }
.cm-tw-syntaxerror { color: rgb(255, 255, 255); background-color: rgb(153, 0, 0); }
.cm-tw-deleted { text-decoration: line-through; }
.cm-tw-header5 { font-weight: 700; }
.cm-tw-listitem:first-child { padding-left: 10px; }
.cm-tw-box { border-style: solid; border-right-width: 1px; border-bottom-width: 1px; border-left-width: 1px; border-color: inherit; border-top-width: 0px !important; }
.cm-tw-underline { text-decoration: underline; }
@media print {
  .CodeMirror div.CodeMirror-cursor { visibility: hidden; }
}


/* noto-sans-sc-300 - latin */
/* noto-sans-sc-300 - latin */
  /* noto-sans-sc-regular - latin */
    /* noto-sans-sc-500 - latin */
    /* noto-sans-sc-700 - latin */
    /* noto-sans-sc-900 - latin */
      

/*** Custom fonts ***/
@import url('file:////home/ninomoriaty/.config/Typora/themes/');

/*** Color setting ***/
:root {
    --side-bar-bg-color: #183055;
    --active-file-bg-color: #2f4566;
    --active-file-text-color: #ffffff;
    --active-file-border-color: #757575;
    --active-search-item-bg-color: #23242b;
    --item-hover-bg-color: #ececec;
    --item-hover-text-color: #f5f7f9;
    --control-text-color: #ddd;
    --window-border: 1px solid #183055;
    --code-cursor: #f0f0f0;
}


/****** #write basic ******/
#write {
    position: static;
    width: 90%;
    max-width: 700px;
    line-height: 1.6;
    transform: none;
    height: auto;
}

/****** #write h1-h6 ******/

#write h1, #write h2, #write h3, #write h4, #write h5, #write h6, #write p, #write pre {
    width: auto;
  }


  
#write h1::before, #write h2::before, #write h3::before, #write h4::before, #write h5::before, #write h6::before {
  position: absolute;
  right: calc(100% + 10px);
  bottom: 0;
  color: #b4b4b4;
  font-size: 1rem;
  font-weight: bold;
  font-variant: 'small-caps';
  border: 0;
  border-radius: 0;
  left: auto;
  float: none;
  padding: 0;
  vertical-align: baseline;
  line-height: 2;
}

#write h1 {
  font-size: 32px;
  font-style: normal;
  font-weight: 800;
  line-height: 2;
  margin-top: 14px;
  margin-bottom: 14px;
}

#write h1::before {
  content: 'H1';
  bottom: 1rem;
  font-weight: bold;
  color: #294f8a;
}

#write h2 {
  font-size: 28px;
  font-weight: 800;
  font-style: normal;
  line-height: 2;
  margin-top: 14px;
  margin-bottom: 14px;
}

#write h2::before {
  content: 'H2';
  bottom: .85rem;
}

#write h3 {
  font-size: 24px;
  font-weight: 800;
  line-height: 2;
  font-style: normal;
  margin-top: 14px;
  margin-bottom: 14px;
}

#write h3::before {
  content: 'H3';
  bottom: .55rem;
}

#write h4 {
  font-size: 20px;
  font-weight: 800;
  line-height: 2;
  font-style: normal;
  margin-top: 14px;
  margin-bottom: 14px;
}

#write h4::before {
  content: 'H4';
  bottom: .35rem;
}

#write h5, #write h6 {
  font-size: 16px;
  font-weight: 800;
  line-height: 2;
  font-style: normal;
  margin-top: 14px;
  margin-bottom: 14px;
}

#write h5::before, #write h6::before {
  bottom: .1rem;
}

#write h5::before {
  content: 'H5';
}

#write h6::before {
  content: 'H6';
}

/****** Global Style ******/
body {
  margin: 0;
  font-family: "Glow Sans SC", -apple-system, sans-serif;
  font-weight: 500;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
  -webkit-text-size-adjust: 100%
}

html, 
body {
  color: #242A31;
  /* width: 100%; */
  height: 100%;
  margin: 0;
  padding: 0;
  font-size: 15px;
  background: #ffffff;
  box-sizing: border-box;
  line-height: 1em;
  text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-text-size-adjust: 100%;
} 

hr {
  border-color: #e6ecf1;
  height: 2px;
  border-top: 2px solid #e6ecf1;
}

img {
  max-width: 80%;
  margin-top: 0.2em;
  margin-bottom: 0.2em;
}

/****** ul ol Style ******/

ul>li>ul>li {
  list-style-type: circle;
  color: #3270cc;
  font-weight: bold;
}

ul>li>ul>li>ul>li {
  list-style-type: square;
  color: #3270cc;
  font-weight: bold;
}

/****** modified 修改过的部分 ******/
ol,
ul {
  padding-left: 2em;
  line-height: 2;
  color: #3270cc;
  font-weight: bold;
}

/****** modified 修改过的部分 ******/
ol>li {
  list-style-type: decimal
}

ol>li>ol>li {
  list-style-type: lower-alpha
}

ol>li>ol>li>ol>li {
  list-style-type: lower-roman
}

/****** Table Style ******/

table {
  padding: 0;
  word-break: initial;
}

table tr {
  border-top: 1px solid #dfe2e5;
  margin: 0;
  padding: 0;
}
table tr:nth-child(2n),
thead {
  background-color: #f5f7f9;
}
table tr th {
  font-weight: bold;
  border: 1px solid #dfe2e5;
  border-bottom: 0;
  margin: 0;
  padding: 6px 13px;
}
table tr td {
  border: 1px solid #dfe2e5;
  margin: 0;
  padding: 6px 13px;
}
table tr th:first-child,
table tr td:first-child {
  margin-top: 0;
}
table tr th:last-child,
table tr td:last-child {
  margin-bottom: 0;
}


/****** YAML Style ******/
pre.md-meta-block {
  font-size: .75rem !important;
  color: #5d5d5d;
  min-height: .8rem;
  white-space: pre-wrap;
  background: #f5f7f9;
  display: block;
  overflow-x: hidden;
  padding: 1rem;
  border-radius: 8px;
}



/****** Global Text ******/
p {
  font-size: 16px;
  font-family: "Glow Sans SC", -apple-system, sans-serif;
  font-weight: 500;
  line-height: 1.6;
  font-style: normal;
  color: rgb(59, 69, 78);
}

a {
  /* color: rgb(164, 78, 237); */
  color: rgb(56, 132, 254);
  font-weight: 500;
  text-decoration: none;
  text-decoration-style: none;
  cursor: pointer;
  padding: 0 3px 0 3px;
}

#write a:hover {
  color: rgb(56, 132, 254);
  text-decoration: underline;
  text-decoration-style: solid;
}

strong {
  font-weight: 700;
}

/****** modified 修改过的部分 ******/
h1 {
  font-size: 2rem;
  font-style: normal;
  font-weight: 800;
  color: #224272;
  line-height: 2;
  margin-top: 14px;
  margin-bottom: 14px;
  border-bottom: 2px solid rgba(204, 204, 204, 0.603);
}

h2 {
  font-size: 1.8px;
  font-weight: 800;
  font-style: normal;
  line-height: 2;
  margin-top: 14px;
  margin-bottom: 14px;
  border-bottom: 1px solid #eee;
}
/****** modified 修改过的部分 ******/

h3 {
  font-size: 1.6px;
  font-weight: 800;
  color: #577097;
  line-height: 2;
  font-style: normal;
  margin-top: 14px;
  margin-bottom: 14px;
}

/****** modified 修改过的部分 ******/
h4 {
  font-size: 1.2px;
  font-weight: 800;
  line-height: 2;
  font-style: normal;
  margin-top: 14px;
  margin-bottom: 14px;
}
/****** modified 修改过的部分 ******/

h5,  h6 {
  font-size: 1rem;
  font-weight: 800;
  line-height: 2;
  font-style: normal;
  margin-top: 14px;
  margin-bottom: 14px;
}


/****** #write Code Fences ******/
#write .md-fences {
    -webkit-font-smoothing: initial;
    margin: 1.8rem 0 2rem 0 !important;
    line-height: 1.43rem;
    border-radius: 3px;
    font-size: 0.95rem;
    word-wrap: normal;
}


#write .CodeMirror-wrap .CodeMirror-code pre {
    padding-left: 12px;
    line-height: 1.55rem;
}
  
#write .CodeMirror-cursors .CodeMirror-cursor {
    border-left: 2px solid var(--code-cursor);
}
  
#write code,
  tt {
    margin: 0 2px;
    padding: 4px 6px;
    border-radius: 6px;
    font-size: 0.92rem;
    background: #f5f7f9;
}
  
#write .md-footnote {
    color: var(--main-5);
    background-color: var(--main-1);
}

.cm-s-inner.CodeMirror, .cm-s-inner .CodeMirror-gutters {
  padding: 0.75rem 0.15rem 0.75rem 0.15rem;
  background-color: #183055 !important;
  color: #f8f8f2 !important;
  border: none;
  border-radius: 6px;
}


.code-tooltip {
  box-shadow: 0 1px 1px 0 rgba(0, 28, 36, .3);
  border-top: 1px solid #eef2f2;
  background: #183055;
  border-radius: 6px;
}

.md-fences {
  font-size: .9rem;
  position: relative!important;
  display: block;
  page-break-inside: avoid;
  text-align: left;
  overflow: visible;
  white-space: pre;
  background: inherit;
}

.md-fences {
  background-color: #f8f8f8;
  margin-bottom: 15px;
  margin-top: 15px;
  padding-top: 8px;
  padding-bottom: 6px;
}

.md-fences, tt {
  border-radius: 3px;
  color: #f0f0f0;
  padding: 0;
  font-size: 0.9em;
}

/****** Sidebar ******/
#typora-sidebar * {
  color: #f0f0f0;
}

#typora-sidebar .file-tree-node.file-library-file-node.active .file-node-background {
  border-left: 5px solid #3884ff;
  height: 2.6rem;
}

.file-list-item {
  border-bottom: var(--window-border);
}

.file-list-item {
  overflow: hidden;
  padding: 12px;
  border-bottom: 1px solid #eee;
  border-bottom: var(--window-border);
  cursor: pointer;
  padding-right: 8px;
  padding-top: 12px;
  padding-left: 24px;
  transition: top .5s;
  -webkit-transition: top .5s;
}

.file-list-item.active {
  background: #2f4566;
  /* background: var(--active-file-bg-color); */
  color: var(--active-file-text-color);
  border-radius: 12px;
}

.file-list-item:not(.active) {
  opacity: .9;
}

.file-node-content {
  padding-top: .8rem;
  margin-bottom: .8rem;
  cursor: default;
  color: var(--control-text-color);
  white-space: nowrap;
  height: 2rem;
}

.file-node-content:active {
  border-radius: 0px!important;
  background: #2f4566;
}

.active .file-node-content {
  font-weight: bold;
}

.file-node-content:hover {
  cursor: pointer;
  border-radius: 0px!important;
}

.file-node-icon, .file-node-open-state {
  display: block;
  float: left;
  line-height: 15px;
  min-height: 15px;
}

.file-node-icon {
  margin-right: 6px;
}

.sidebar-footer {
background: var(--side-bar-bg-color);
border-top: 1px #555 solid;
}

#typora-sidebar #ty-sidebar-footer .sidebar-footer-item:hover {
background: #021d43;
}

#typora-sidebar #outline-content .outline-item:hover {
background: #202020;
}

#typora-sidebar #ty-sidebar-footer #sidebar-files-menu .show + .menuitem-group-label.show {
border-color: #202020;
}

#typora-sidebar #ty-sidebar-footer {
border-top: 1px solid #19191c;
}

#typora-sidebar #ty-sidebar-footer #sidebar-files-menu li > a:hover {
background: #021d43;
}

#typora-sidebar #ty-sidebar-footer #sidebar-files-menu .ty-side-sort-btn.active, #typora-sidebar #ty-sidebar-footer #sidebar-files-menu .ty-side-sort-btn:hover {
color: #3884FF;
}

#typora-sidebar #ty-sidebar-footer #sidebar-files-menu .ty-side-sort-btn.active {
background: #001129;
}

#typora-sidebar .file-list-item.file-library-node:not(.active):hover {
  background: #243959;
  border-radius: 12px;
}

#typora-sidebar .file-tree-node.file-library-file-node:not(.active):hover .file-node-background {
  background: #243959;
  border-radius: 12px;
  height: 2.6rem;
}

/****** Quote style ******/
blockquote {
    position: relative;
  /*  margin: 1rem 1 1rem 2rem; */
    padding: 1rem;
    color: #827676;
    background-color: #f5f7f9;
    border-radius: 6px;
    line-height: 1;
}
  
blockquote::before {
    content: '';
    position: absolute;
    top: 0rem;
    left: 0rem;
    height: 100%;
    width: .30rem;
    background: #3884ff;
    border-top-left-radius: 6px;
    border-bottom-left-radius: 6px;
}



/****** Code style ******/
code {
    font-size: .9em; 
  /*   color: #333333; /*  #537AA2;  
      border: 1px solid #d0d0d0;  */
    font-family: 'Cascadia Code', Consolas, 'Noto Sans SC','Courier New', monospace;
    padding: .2em .2em;
    border-radius: 3px;
    background: #f5f7f9 !important;
}
  
code,
pre{
    font-size: 90% !important;
    font-weight: normal;
    font-family: 'Cascadia Code', Consolas, 'Noto Sans SC','Courier New', monospace;
    -webkit-font-smoothing: initial;
    -moz-osx-font-smoothing: initial
}

/****** Code highlight ******/
.cm-s-inner .CodeMirror-guttermarker, .cm-s-inner .CodeMirror-guttermarker-subtle, .cm-s-inner .CodeMirror-linenumber { color: #596774; }
.cm-s-inner .CodeMirror-cursor { border-left: 1px solid #f8f8f0; }
.cm-s-inner div.CodeMirror-selected { background: rgba(255, 255, 255, 0.15); }
.cm-s-inner.CodeMirror-focused div.CodeMirror-selected { background: rgba(255, 255, 255, 0.10); }
.cm-s-inner .CodeMirror-line::selection, .cm-s-inner .CodeMirror-line > span::selection, .cm-s-inner .CodeMirror-line > span > span::selection { background: rgba(255, 255, 255, 0.10); }
.cm-s-inner .CodeMirror-line::-moz-selection, .cm-s-inner .CodeMirror-line > span::-moz-selection, .cm-s-inner .CodeMirror-line > span > span::-moz-selection { background: rgba(255, 255, 255, 0.10); }

.cm-s-inner .CodeMirror-activeline-background { background: rgba(0, 0, 0, 0); }
.cm-s-inner .cm-keyword { color: rgba(199, 146, 234, 1); }
.cm-s-inner .cm-operator { color: rgba(233, 237, 237, 1); }
.cm-s-inner .cm-variable-2 { color: #80CBC4; }
.cm-s-inner .cm-variable-3 { color: #82B1FF; }
.cm-s-inner .cm-builtin { color: #DECB6B; }
.cm-s-inner .cm-atom { color: #F77669; }
.cm-s-inner .cm-number { color: #F77669; }
.cm-s-inner .cm-def { color: rgba(233, 237, 237, 1); }
.cm-s-inner .cm-string { color: #C3E88D; }
.cm-s-inner .cm-string-2 { color: #80CBC4; }
.cm-s-inner .cm-comment { color: #546E7A; }
.cm-s-inner .cm-variable { color: #82B1FF; }
.cm-s-inner .cm-tag { color: #80CBC4; }
.cm-s-inner .cm-meta { color: #80CBC4; }
.cm-s-inner .cm-attribute { color: #FFCB6B; }
.cm-s-inner .cm-property { color: #80CBAE; }
.cm-s-inner .cm-qualifier { color: #DECB6B; }
.cm-s-inner .cm-variable-3 { color: #DECB6B; }
.cm-s-inner .cm-tag { color: rgba(255, 83, 112, 1); }
.cm-s-inner .cm-error {
  color: rgba(255, 255, 255, 1.0);
  background-color: #EC5F67;
}
.cm-s-inner .CodeMirror-matchingbracket {
  text-decoration: underline;
  color: white !important;
}

.md-fences .code-tooltip {
  background-color: #263238;
}




</style><title>BioIT and DataSt summary</title>
</head>
<body class='typora-export'><div class='typora-export-content'>
<div id='write'  class=''><p><span>Guideline Level</span></p><table><tbody><tr><td style="background: #3ADDBA !important;"><center><font style="font-size:45px; color:black !important;"><b>
    Bioinformatics Toolkits and Data Structure in Bioinformatic
    </b></font></center></td></tr></tbody></table><h1 id='ref'><span>Ref</span></h1><p><span>Tips: Needed to be assigned to corresponding  Section</span></p><ol start='' ><li><span>(NCBI) BLAST User guide: </span><a href='https://www.ncbi.nlm.nih.gov/books/NBK279690/' target='_blank' class='url'>https://www.ncbi.nlm.nih.gov/books/NBK279690/</a></li><li><span>Another BLAST Usage Notebook: </span><a href='https://www.ccg.unam.mx/~vinuesa/tlem/pdfs/Bioinformatics_explained_BLAST.pdf' target='_blank' class='url'>https://www.ccg.unam.mx/~vinuesa/tlem/pdfs/Bioinformatics_explained_BLAST.pdf</a></li></ol><p>&nbsp;</p><h1 id='lk'><span>Lk</span></h1><ul><li><span>Lk: common workflow will be stored in this folder and detailed records and manual can be found in the Wiznote</span></li><li><span>E:\0 Science\Biology\QB Quantitative Biology-计量生物学\BioI-生信\BioIToolkit\NCBI\BLAST</span></li></ul><p>&nbsp;</p><h1 id='pcp-main'><span>Pcp-Main</span></h1><h2 id='bioinformatic-toolkit-section'><span>Bioinformatic Toolkit Section</span></h2><ul><li><p><span>This note book summarise all bioinformatic tools for different usage and provide proper Lk for further manual or usage.</span></p><ul><li><span>Thus this note book focus on complete collection and reference guideline rather than detailed usage.</span></li><li><span>More information should be found through Lk.</span></li></ul></li><li><p><span>To simplify the and make the search for particular tool more easily, this notebook only contain: </span></p><ul><li><span>Ref: The website for download, documents and further information</span></li><li><span>Lk: The Wiznote note for further usage</span></li><li><span>F: The main functions of the tool</span></li><li><span>Mt-Install: The basic installation method</span></li><li><span>Tips: Reminder for some preference settings or prerequisites.</span></li></ul></li><li><p><span>The pipelines for different stages can be found in Lk notes and you should check them in corresponding Wiznote note.</span></p></li></ul><p>&nbsp;</p><h2 id='database-in-bioi-section'><span>Database in BioI Section</span></h2><ul><li><span>This note book also contain the collection of bioinformatic database for different function.</span></li><li></li></ul><p>&nbsp;</p><h2 id='data-structure-in-bioi-section'><span>Data Structure in BioI Section</span></h2><ul><li><span>Talking of database, the bioinformatic-specific data structures will also be introduced in this note book and you can find the </span></li></ul><p>&nbsp;</p><h2 id='high-performance-computation-section'><span>High-Performance Computation Section</span></h2><ul><li><span>This part introduce some usage of the high performance machine during my learning experience.</span></li></ul><p>&nbsp;</p><p>&nbsp;</p><table><tbody><tr><td style="background: #3ADDBA !important;"><center><font style="font-size:45px; color:black !important;"><b>
    Bioinformatics Toolkits
    </b></font></center></td></tr></tbody></table><h1 id='1-bioinformatics-toolkits'><span>1 Bioinformatics Toolkits</span></h1><h1 id='knowledge-mapping-for-papers'><span>Knowledge Mapping for papers</span></h1><h3 id='citespace'><span>Citespace</span></h3><p>&nbsp;</p><h1 id='synthetic-circuit-design'><span>Synthetic Circuit Design</span></h1><h3 id='igem-cello'><span>iGEM cello</span></h3><p>&nbsp;</p><h1 id='wet-experiment-manager'><span>Wet Experiment Manager</span></h1><h2 id='integrated'><span>Integrated</span></h2><h3 id='vector-nti-express-designer'><span>Vector NTI Express Designer</span></h3><p>&nbsp;</p><h2 id='primeradaptor-design'><span>Primer/Adaptor Design</span></h2><h3 id='primer-premier'><span>Primer Premier</span></h3><p>&nbsp;</p><h1 id='data-download-and-data-conversion'><span>Data Download and Data Conversion</span></h1><h3 id='sratoolkit'><span>SRAToolkit</span></h3><p><strong><span>Ref</span></strong></p><ul><li><span>GitHub Documentation:</span><a href='https://github.com/ncbi/sra-tools/wiki/02.-Installing-SRA-Toolkit' target='_blank' class='url'>https://github.com/ncbi/sra-tools/wiki/02.-Installing-SRA-Toolkit</a></li><li></li></ul><p><strong><span>Obj1+2</span></strong></p><ul><li><span>Datasets from NCBI database</span></li><li></li></ul><p><strong><span>Mt-faster-dump</span></strong></p><p>&nbsp;</p><p><strong><span>F</span></strong></p><ul><li><span>Standard method to download data from NCBI</span></li><li><span>sequence data conversion</span></li><li></li></ul><h3 id='ispemra'><span>ISPEMRA</span></h3><p><strong><span>Func</span></strong></p><ul><li><span>Alternative/Integrated download method for NCBI and EBI data.</span></li></ul><h3 id='seqtk'><span>seqtk</span></h3><p><strong><span>Func</span></strong></p><ul><li><span>Convert fastq to fasta</span></li><li><span>parse and filter sequence data with hard conditions</span></li><li></li></ul><p>&nbsp;</p><h1 id='sequencing-and-assembly'><span>Sequencing and Assembly</span></h1><h2 id='k-mers-and-k-mer-spectra'><span>k-mers and k-mer spectra</span></h2><h3 id='jellyfish'><span>Jellyfish</span></h3><p>&nbsp;</p><h3 id='kmc'><span>kmc</span></h3><p>&nbsp;</p><h3 id='kat'><span>KAT</span></h3><p>&nbsp;</p><p>&nbsp;</p><h2 id='rna-seq'><span>RNA-Seq</span></h2><h3 id='rascaf'><span>Rascaf</span></h3><p><span>Ref: </span><a href='https://github.com/mourisl/Rascaf' target='_blank' class='url'>https://github.com/mourisl/Rascaf</a></p><h3 id='l-rnascaffolder'><span>L-RNA_SCAFFOLDER</span></h3><p><span>Ref: </span><a href='https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-14-604' target='_blank' class='url'>https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-14-604</a></p><h3 id='scubat'><span>SCUBAT</span></h3><p><span>Ref: </span><a href='https://github.com/GDKO/SCUBAT2' target='_blank' class='url'>https://github.com/GDKO/SCUBAT2</a></p><h3 id='rnapath-in-erange'><span>RNAPATH in ERANGE</span></h3><p><span>Ref: </span><a href='https://genome.cshlp.org/content/20/12/1740.full.html' target='_blank' class='url'>https://genome.cshlp.org/content/20/12/1740.full.html</a></p><p>&nbsp;</p><h2 id='sequence-alignment'><span>Sequence Alignment</span></h2><h3 id='blast'><span>BLAST</span></h3><p><strong><span>Ref</span></strong><span>:</span></p><ol start='' ><li><span>Web-based toolkit: </span><a href='https://blast.ncbi.nlm.nih.gov/Blast.cgi' target='_blank' class='url'>https://blast.ncbi.nlm.nih.gov/Blast.cgi</a></li></ol><p><strong><span>Soc</span></strong></p><ol start='' ><li><span>NCBI</span></li></ol><p><strong><span>Def</span></strong><span>: identify  putatively homologous sequences in a database</span></p><p><strong><span>Obj2</span></strong><span>:</span></p><ol start='' ><li><span>DNA sequences</span></li><li><span>Protein sequences</span></li><li><span>Phylogenetic tree groups</span></li></ol><p><strong><span>Ab</span></strong><span>:</span></p><ol start='' ><li><p><span>Data Ab:</span></p><ol start='' ><li><p><span>DNA</span></p><ol start='' ><li><span>nucleotide sequences</span></li></ol></li><li><p><span>Protein</span></p><ol start='' ><li><span>Amino acid sequences</span></li><li><span>Distance in genetic code and components from different genes in the genome will make the protein sequences hard to match</span></li><li><span>The chemical similarity in some amino acid and protein structures make the protein alignment more challenging.</span></li><li><span>Existing alignments and structures may not suit the new structures of the sequencing proteins.</span></li></ol></li></ol></li></ol><p><strong><span>Mt-Install</span></strong></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">sudo</span> apt-get install ncbi-blast<span class="cm-operator">+</span></span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 23px;"></div><div class="CodeMirror-gutters" style="display: none; height: 23px;"></div></div></div></pre><p><strong><span>Mt-BLAST process</span></strong><span>:</span></p><ol start='' ><li><u><span>Build an index</span></u><span> of short words for all sequences in a database</span></li><li><u><span>Search matching short words</span></u><span> in the query sequences  </span></li><li><span>Extend each match to the left or the right to </span><u><span>make the longest match</span></u><span> (High scoring Segment Pair, HSP)</span></li><li><span>Stop the extending when the match score does go up</span></li><li><span>Try to merge HSPs to get even longer matches</span></li></ol><p><strong><span>Mt-Use BLAST in command line</span></strong></p><p><span>    Ref: </span><a href='https://www.ncbi.nlm.nih.gov/books/NBK25501/' target='_blank' class='url'>https://www.ncbi.nlm.nih.gov/books/NBK25501/</a></p><ol start='' ><li><p><span>build the BLAST database</span></p><ol start='' ><li><p><span>Select proper database</span></p><ol start='' ><li><p><span>NCBI database list: </span><a href='https://www.ncbi.nlm.nih.gov/guide/all/' target='_blank' class='url'>https://www.ncbi.nlm.nih.gov/guide/all/</a></p><ol start='' ><li><span>nr : non-redundant protein database</span></li><li><span>nt : non-redundant nucleotide database</span></li><li><span>swiss-prot : manually annotated protein database</span></li><li><span>pat : protein sequences from patents</span></li><li><span>month : recently added sequences</span></li><li><span>env_nr : environmental sequences</span></li><li><span>est_human : only human Expressed Sequence Tag sequences</span></li></ol></li><li><p><span>Access the database remotely when the database is relatively large</span></p></li></ol></li><li><p><span>Get the datasets from the database</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># check the workplace first</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">pwd</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># download the datasest</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">wget</span> http://nematodes.org/teaching/bioinformatics/nem.fasta</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># get particular database txt from the html</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">wget</span> <span class="cm-attribute">-O</span> testsequence.fasta <span class="cm-string">"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nuccore&amp;id=NC_015119.1&amp;strand=1&amp;seq_start=1426&amp;seq_stop=2962&amp;rettype=fasta&amp;retmode=text"</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Explanation</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">-O</span> testsequence.fasta (the Output file)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">efetch.fcgi? (there are many NCBI utilities, described <span class="cm-keyword">in</span> more detail at https://www.ncbi.nlm.nih.gov/books/NBK25501/)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">db</span><span class="cm-operator">=</span>nuccore (which database)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">id</span><span class="cm-operator">=</span>NC_015119.1 (sequence AccessionID)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">strand</span><span class="cm-operator">=</span><span class="cm-number">1</span> (top strand)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">seq_start</span><span class="cm-operator">=</span><span class="cm-number">1426</span> (start from base <span class="cm-number">1426</span> of the sequence)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">seq_stop</span><span class="cm-operator">=</span><span class="cm-number">2962</span> (stop at base <span class="cm-number">2962</span> of the sequence)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">rettype</span><span class="cm-operator">=</span>fasta (return <span class="cm-keyword">in</span> fasta format)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">retmode</span><span class="cm-operator">=</span>text (return <span class="cm-keyword">in</span> plain text format please!)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 529px;"></div><div class="CodeMirror-gutters" style="display: none; height: 529px;"></div></div></div></pre></li><li><p><span>Inspect the data and see if the data is correct</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">cat</span> nem.fasta | <span class="cm-builtin">awk</span> <span class="cm-string">'{</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">    </span><span class="cm-comment"># Is this line a fasta header</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">    </span><span class="cm-keyword">if</span>(substr(<span class="cm-def">$1</span>,1,1)<span class="cm-operator">==</span><span class="cm-string">"&gt;"</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">    </span>{</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">    </span> print <span class="cm-string">"this is a header line: "</span> <span class="cm-def">$0</span>; </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">    </span> <span class="cm-def">hline</span><span class="cm-operator">=</span>FNR ;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">    </span>}</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">    </span><span class="cm-comment"># Identify the first line of the sequence</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">    </span><span class="cm-comment"># ...and then the first character of that line</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">    </span><span class="cm-keyword">if</span><span class="cm-def">(FNR</span><span class="cm-operator">==</span>hline<span class="cm-operator">+</span><span class="cm-number">1</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">    </span>{</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">    </span> print <span class="cm-string">"First line is:"</span> <span class="cm-def">$0</span>;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; <span class="cm-def">first_seq_character</span><span class="cm-operator">=</span>substr(<span class="cm-def">$0</span>,1,1)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; print <span class="cm-string">"First character is: "</span> first_seq_character</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; print first_seq_character &gt; <span class="cm-string">"first_seq_character.txt"</span> </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  }</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">}<span class="cm-string">'</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 402px;"></div><div class="CodeMirror-gutters" style="display: none; height: 402px;"></div></div></div></pre><p><span>Normally, FASTA data will have the structure described in the Data Structure in BioI part in this notebook.</span></p></li><li><p><span>Build the database structure that suit the usage of blast analysis</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># help manual</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">makeblastdb <span class="cm-attribute">--help</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Common usage</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">makeblastdb <span class="cm-attribute">-in</span> nem.fasta <span class="cm-attribute">-dbtype</span> prot <span class="cm-attribute">-out</span> nem</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># -in the input of dataset</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># -dbtype the type of database</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># -out the output NAME of the database for blast</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Input</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">nem.fasta: database input </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Output</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">nem.phr: protein sequences headers(Names of the protein and basic information)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">nem.pin: protein index(The locations of the short words)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">nem.psq: compressed protein sequences</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 368px;"></div><div class="CodeMirror-gutters" style="display: none; height: 368px;"></div></div></div></pre></li><li><p><span>Get the query sequences</span></p><p><span>From a summary accession list provided by NCBI:</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-keyword">while</span> read acc</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-keyword">do</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">echo</span> <span class="cm-attribute">-e</span> <span class="cm-string">"Downloading </span><span class="cm-def">$acc</span><span class="cm-string">"</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">wget</span> <span class="cm-attribute">-q</span> <span class="cm-attribute">-O</span> <span class="cm-def">$acc</span>.fasta <span class="cm-string">"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=protein&amp;id=</span><span class="cm-def">$acc</span><span class="cm-string">&amp;strand=1&amp;rettype=fasta&amp;retmode=text"</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-keyword">done</span> &lt; 590accessions.txt </span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 161px;"></div><div class="CodeMirror-gutters" style="display: none; height: 161px;"></div></div></div></pre><p><span>Actually, </span><code>efetch</code><span> will be a better choice</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 23px;"></div><div class="CodeMirror-gutters" style="display: none; height: 23px;"></div></div></div></pre></li><li><p>&nbsp;</p></li></ol></li><li><p><span>Search and match targeted sequences</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># DNA - DNA: </span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">blastn</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">megablast</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">tblastx</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># DNA - Protein</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">blastx <span class="cm-attribute">-db</span> nem <span class="cm-attribute">-query</span> testsequence.fasta</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">blastx <span class="cm-attribute">-db</span> nem <span class="cm-attribute">-query</span> testsequence.fasta <span class="cm-attribute">-outfmt</span> <span class="cm-number">7</span> &gt; blastoutput1.out</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># -db: database</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># -query: sequences needed to match to the database</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># -outfmt 7: the output format tsv format</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Protein - DNA</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">tblastn</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Protein -Protein</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">blastp</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">quickblastp</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">psi-blast</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 460px;"></div><div class="CodeMirror-gutters" style="display: none; height: 460px;"></div></div></div></pre><p><strong><span>Mt-Choose proper methods for pairs of query and database</span></strong></p><p><span>Tips: The first is the query and the second is the database</span></p><ol start='' ><li><span>DNA - DNA: blastn, megablast, tblastx, </span><a href='https://ncbi.github.io/magicblast/'><span>Magic BLAST</span></a></li><li><span>DNA - Protein: blastx</span></li><li><span>Protein - DNA: tblastn</span></li><li><span>Protein -Protein: blastp, quickblastp, psi-blast</span></li></ol></li><li><p><span>Inspect and interpret the results</span></p><ol start='' ><li><p><strong><span>Comment</span></strong><span> lines start with #</span></p><ol start='' ><li><strong><span>Fields</span></strong><span> in the comments section are separated by a space</span></li></ol></li><li><p><strong><span>Data Section</span></strong></p><ol start='' ><li><span>each data section is a HSP</span></li><li><span>separated by tab or the output format</span></li></ol></li><li><p>&nbsp;</p></li></ol></li><li><p>&nbsp;</p></li><li><p>&nbsp;</p></li></ol><p><strong><span>Mt-Bitscore Calculation</span></strong></p><ol start='' ><li><p><span>Ab:</span></p><ol start='' ><li><p><span>The scoring matrix</span></p><ol start='' ><li><a href='http://www.drive5.com/usearch/manual/karlin_altschul.html'><span>Karlin-Altschul statistics</span></a><span> are used to convert the HSP bitscore to an expectation value (E-value) </span></li><li><span>bigger HSP bitscore are &quot;better&quot; and smaller E-values are &quot;better&quot; </span></li><li><span>not all HSP are equally comparable</span></li></ol></li><li><p><span>The length of query</span></p></li><li><p><span>the word size</span></p></li><li><p><span>the size of the database</span></p></li><li><p>&nbsp;</p></li></ol></li></ol><p><strong><span>Mt-Scoring matrix</span></strong></p><p><span>Ref: </span><a href='https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3848038/'><span>Selecting the Right Similarity-Scoring Matrix</span></a></p><p><span>DNA</span></p><ol start='' ><li><span>The simplest example: just count the match number and give it 1 value and the unmatched nucleotide get 0 value.</span></li><li></li></ol><p><span>Protein</span></p><ol start='' ><li><p><strong><span>p</span></strong><span>oint </span><strong><span>a</span></strong><span>ccepted </span><strong><span>m</span></strong><span>utation (</span><strong><span>PAM</span></strong><span>): the replacement of a single amino acid in the primary structure of a protein with another single amino acid, which is accepted by the processes of natural selection based on global (end to end) alignments of highly similar proteins.</span></p><p><img src="https://i.loli.net/2020/11/14/PNIgrQAHKZYtXaV.png" referrerpolicy="no-referrer" alt="Screenshot from 2020-11-13 19-55-40.png"></p><ol start='' ><li><p><span>Cat:</span></p><ol start='' ><li><span>PAM N corresponds to </span><u><span>N mutations in DNA sequence</span></u><span> per 100 amino acids. N can be &gt; 100</span></li><li><span>PAM250 is one of the more common used scoring matrices.</span></li></ol></li><li><p><span>Ab/Pcp-PAM model naming convention: </span></p><ol start='' ><li><span>high number for distantly related proteins, low number for closely related protein </span></li></ol></li></ol></li><li><p><strong><span>BLOSUM matrices</span></strong><span>: BLOSUM (</span><strong><span>BLO</span></strong><span>cks </span><strong><span>SU</span></strong><span>bstitution </span><strong><span>M</span></strong><span>atrix) matrix is a substitution matrix used for scoring sequence alignment of </span><u><span>evolutionary divergent</span></u><span> protein sequences based on local alignments.</span></p><p><img src="https://i.loli.net/2020/11/14/DeAm4h6VLpMzgNq.png" referrerpolicy="no-referrer" alt="Screenshot from 2020-11-13 19-55-46.png"></p><ol start='' ><li><p><span>Ref: </span><a href='https://www.pnas.org/content/89/22/10915.long' target='_blank' class='url'>https://www.pnas.org/content/89/22/10915.long</a></p></li><li><p><span>Ab</span></p><ol start='' ><li><span>All BLOSUM matrices are based on observed alignments; they are not extrapolated from comparisons of closely related proteins like the PAM Matrices.</span></li></ol></li><li><p><span>Mt</span></p><ol start='' ><li><span>scanned the BLOCKS database for very conserved regions of protein families(UNGAPPED local alignments)</span></li><li><span>counted the relative frequencies of amino acids and their substitution probabilities.</span></li><li></li></ol></li><li><p><span>Cat</span></p><ol start='' ><li><span>BLOSUM62 is the matrix built using sequences with </span><u><span>no more than 62% similarity</span></u><span>; BLOSUM62 is the </span><u><span>default matrix used for protein BLAST</span></u><span>.</span></li><li><span>BLOSUM80 is for less divergent alignments, whereas BLOSUM 50 is for more divergent alignments.</span></li></ol></li></ol></li></ol><p><strong><span>F</span></strong><span>:</span></p><ol start='' ><li><span>Predict statistical evolution relationship but not the certain relationship</span></li><li><span>Pairwise alignment</span></li><li><u><span>Functional identification</span></u><span> of novel sequences</span></li><li><u><span>annotation</span></u><span> of genomes</span></li><li><u><span>localisation</span></u><span> of particular sequences in </span><u><span>genomes</span></u></li><li><span>screening of </span><u><span>environments</span></u><span> using set of sequences</span></li><li><u><span>clustering</span></u><span> of sequence data to identify sequence families</span></li><li><span>To be continued...</span></li></ol><p>&nbsp;</p><h3 id='clustalo'><span>Clustalo</span></h3><p><strong><span>Ref</span></strong><span>: </span><a href='http://www.ebi.ac.uk/Tools/msa/clustalo/' target='_blank' class='url'>http://www.ebi.ac.uk/Tools/msa/clustalo/</a></p><p><strong><span>Source</span></strong><span>: EBI</span></p><p><strong><span>F</span></strong><span>:</span></p><ol start='' ><li><span>You can use the </span><code>Clustalo Omega</code><span> website toolkit to align the sequences online. The result and summary will be valid for 7 days.</span></li><li></li></ol><p><strong><span>Cat</span></strong><span>:</span></p><ol start='' ><li><span>Clustalo</span></li><li><span>Clustalw</span></li></ol><p><strong><span>Mt-install</span></strong></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">sudo</span> apt install clustalo</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 23px;"></div><div class="CodeMirror-gutters" style="display: none; height: 23px;"></div></div></div></pre><h3 id='bowtie2'><span>bowtie2</span></h3><h3 id='bwa'><span>bwa</span></h3><p><strong><span>F</span></strong></p><ul><li><span>Read mapper</span></li></ul><h3 id='hisat2'><span>hisat2</span></h3><h3 id='picard'><span>picard</span></h3><h3 id='igv'><span>IGV</span></h3><p><strong><span>Def</span></strong></p><ul><li><span>integrated genome viewer</span></li></ul><p><strong><span>F</span></strong></p><ul><li></li></ul><p>&nbsp;</p><h2 id='sequencing-quality-controlqc'><span>Sequencing Quality Control(QC)</span></h2><p><font color=orange size=6><span>Quality Assessment</span></font></p><h3 id='fastqc'><span>FastQC</span></h3><p><strong><span>Ref</span></strong></p><ul><li><span>Website </span><a href='https://www.bioinformatics.babraham.ac.uk/projects/fastqc/' target='_blank' class='url'>https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a></li></ul><p><strong><span>Lk</span></strong></p><ul><li><span>Further usage in sequencing assessment: Wiznote &gt; 0 Science &gt; QB-计量生物学 &gt; BioI-生信 &gt; BioIToolkit &gt; Sequencing Analysis Research</span></li></ul><p><strong><span>Mt-Install from zip file</span></strong></p><ul><li><p><span>Download the zip file from the website</span></p></li><li><p><span>Extract the folder from the zip file</span></p></li><li><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Access the fastqc program</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">chmod</span> <span class="cm-number">755</span> ./fastqc</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Settle the PATH for fastqc in .bashrc</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-keyword">export</span> <span class="cm-def">PATH</span><span class="cm-operator">=</span>~/[Your dir]/FastQC:<span class="cm-def">$PATH</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Try</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">fastqc</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 138px;"></div><div class="CodeMirror-gutters" style="display: none; height: 138px;"></div></div></div></pre></li></ul><p><strong><span>Mt-Install from the pkg store</span></strong></p><ul><li><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># From apt</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">sudo</span> apt install fastqc</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Frome conda</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">conda install</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 92px;"></div><div class="CodeMirror-gutters" style="display: none; height: 92px;"></div></div></div></pre></li></ul><h3 id='nanoplot'><span>NanoPlot</span></h3><p><strong><span>Ref</span></strong></p><p><strong><span>Obj1</span></strong></p><ul><li></li></ul><p><strong><span>F</span></strong></p><ul><li><span>Visualise the Quality Assessment of Nanopore genome data</span></li></ul><p>&nbsp;</p><hr /><p><font color=orange size=6><span>Quality Improvement</span></font></p><h3 id='cutadapt'><span>Cutadapt</span></h3><p><strong><span>Ref</span></strong></p><p><a href='https://cutadapt.readthedocs.io/en/stable/' target='_blank' class='url'>https://cutadapt.readthedocs.io/en/stable/</a></p><p><strong><span>F</span></strong></p><ul><li></li></ul><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><h2 id='genome-assembly'><span>Genome Assembly</span></h2><p><font color=orange size=6><span>Short Read</span></font></p><h3 id='quiver'><span>Quiver</span></h3><p><span>Ref: </span><a href='https://github.com/PacificBiosciences/GenomicConsensus' target='_blank' class='url'>https://github.com/PacificBiosciences/GenomicConsensus</a></p><p><span>F: Uses shorter PB reads to correct longer ones</span></p><p><span>RSI, RSII early </span></p><h3 id='arrow'><span>Arrow</span></h3><p><span>Ref: </span><a href='https://github.com/PacificBiosciences/GenomicConsensus' target='_blank' class='url'>https://github.com/PacificBiosciences/GenomicConsensus</a></p><p><span>RSII and SEQUEL</span></p><h3 id='nanopolish'><span>Nanopolish</span></h3><p><span>Ref: </span><a href='https://github.com/jts/nanopolish' target='_blank' class='url'>https://github.com/jts/nanopolish</a></p><p><span>F: Nanopore only; uses original voltage data to correct errors</span></p><hr /><p><font color=orange size=6><span>Long Read</span></font></p><h3 id='pbjelly'><span>PBJelly</span></h3><p><span>Cat</span></p><h3 id='juicebox'><span>Juicebox</span></h3><p><span>Cat+F: HiC Contact Map</span></p><h3 id='hierarchical-genome-assembly-process-hgap'><span>Hierarchical Genome Assembly Process (HGAP)</span></h3><p><span>Ref: </span><a href='https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/HGAP' target='_blank' class='url'>https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/HGAP</a></p><p><span>Pcp</span></p><ul><li><span>Use the long reads to construct pre-assembled reads (longest) as corrected seeds</span></li><li><span>Then use the pre-assembled reads to assemble the finished genome</span></li></ul><p><span>Cat+F: </span></p><ul><li><span>Long Read Assembly</span></li><li><span>reduce the influence of relative short reads by using pre-assembly method.</span></li></ul><h3 id='canu'><span>Canu </span></h3><p><span>Ref: </span><a href='https://github.com/marbl/canu' target='_blank' class='url'>https://github.com/marbl/canu</a></p><p><span>Obj1: PacBio or Nanopore</span></p><p><span>Pcp</span></p><ul><li><span>OLC</span></li></ul><p><span>Pc</span></p><ol start='' ><li><span>Correct</span></li><li><span>Trim</span></li><li><span>OLC Assembly</span></li></ol><p><span>Cat+F: </span></p><ul><li><span>Long Read Assembly</span></li><li></li></ul><h3 id='falcon--falcon-unzip'><span>FALCON / FALCON-unzip</span></h3><p><span>Ref: </span><a href='https://github.com/PacificBiosciences/FALCON' target='_blank' class='url'>https://github.com/PacificBiosciences/FALCON</a></p><p><span>Obj1: PacBio</span></p><p><span>Cat+F: </span></p><ul><li><span>Long Read Assembly</span></li><li><span>Show the haplotype-resolved assembly graph and the SNPs and SV situation for the assembly. </span></li></ul><h3 id='miniasm'><span>Miniasm </span></h3><p><span>Ref: </span><a href='https://github.com/lh3/miniasm' target='_blank' class='url'>https://github.com/lh3/miniasm</a></p><p><span>Obj1: PacBio and Nanopore</span></p><p><span>Pcp</span></p><ul><li><span>OLC</span></li><li><span>Use k-mers minimisers to allow fast matching and allow low Q sequence</span></li></ul><p><span>Ab</span></p><ul><li><span>Able to deal with noisy long read data</span></li><li><span>No error correction</span></li><li><span>Fast</span></li><li></li></ul><p><span>Cat+F: Long Read Assembly</span></p><h3 id='wtdbg2redbean'><span>wtdbg2/RedBean</span></h3><p><span>Ref: </span><a href='https://github.com/ruanjue/wtdbg' target='_blank' class='url'>https://github.com/ruanjue/wtdbg</a></p><p><span>Pcp</span></p><ul><li><span>De Bruijn Graph(DBG) approach</span></li><li><span>Use FBG of K-bins(a K-bin is 256 bases)</span></li><li><span>Polish after assembly</span></li></ul><p><span>Ab</span></p><ul><li><span>High quality</span></li></ul><p><span>Pc</span></p><ol start='' ><li><span>Binning the reads</span></li><li><span>Match the overlaps of each reads&#39; k-bin and construct the DBG</span></li><li><span>Clean the DBG</span></li><li><span>Consensus</span></li></ol><p><span>Cat+F: Long Read Assembly</span></p><p>&nbsp;</p><h2 id='genome-assembly-quality-control'><span>Genome Assembly Quality Control</span></h2><p><font color=orange size=6><span>Quality Assessment</span></font></p><h3 id='busco'><span>BUSCO</span></h3><p><strong><span>Ref</span></strong></p><ul><li><span>GitLab: </span><a href='https://gitlab.com/ezlab/busco' target='_blank' class='url'>https://gitlab.com/ezlab/busco</a></li><li><span>Documentation: </span><a href='https://busco.ezlab.org/busco_userguide.html' target='_blank' class='url'>https://busco.ezlab.org/busco_userguide.html</a></li></ul><p>&nbsp;</p><h3 id='cegma'><span>CEGMA</span></h3><p><strong><span>Tips</span></strong></p><ul><li><span>This benchmark was discontinued from 2015 and you should choose BUSCO to do the analysis</span></li></ul><p><strong><span>Ref</span></strong></p><ul><li></li></ul><h3 id='assembly-stats'><span>assembly-stats</span></h3><p><strong><span>Ref</span></strong></p><ul><li><span>GitHub: </span><a href='https://github.com/rjchallis/assembly-stats' target='_blank' class='url'>https://github.com/rjchallis/assembly-stats</a></li><li><span>Website and Documentation: </span><a href='https://assembly-stats.readme.io/docs' target='_blank' class='url'>https://assembly-stats.readme.io/docs</a></li></ul><p><strong><span>Tips</span></strong></p><ul><li><span>This project is not yet completed and the documentation can be the worst manual for me.</span></li></ul><p><strong><span>Func</span></strong></p><ul><li><span>Visualise the statistics of contigs in genome assembly</span></li></ul><p><strong><span>Mt-Install</span></strong></p><ul><li><p><span>For GitHub</span></p><ul><li><span>Download repo from the GitHub</span></li><li><span>Set up the PATH for pl folder of the repo in .bashrc</span></li></ul></li></ul><p><strong><span>Mt-Usage</span></strong></p><ul><li><p><span>First, use the Perl Script in the pl folder of the repo to generate json object</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Remember the json object should have the .assembly-stats.json for HTML usage</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">perl asm2stats.pl genome_assembly.fa &gt; output.assembly-stats.json</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 46px;"></div><div class="CodeMirror-gutters" style="display: none; height: 46px;"></div></div></div></pre><p>&nbsp;</p></li><li><p><span>Then insert the json object into the HTML file of the repo (Tips: The json)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="html"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="html"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tag cm-bracket">&lt;</span><span class="cm-tag">div</span> <span class="cm-attribute">id</span>=<span class="cm-string">"assembly_stats"</span><span class="cm-tag cm-bracket">&gt;</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tag cm-bracket">&lt;</span><span class="cm-tag">script</span><span class="cm-tag cm-bracket">&gt;</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">d3</span>.<span class="cm-property">json</span>(<span class="cm-string">"Danaus_plexippus_v3.assembly-stats.json"</span>, <span class="cm-keyword">function</span>(<span class="cm-def">error</span>, <span class="cm-def">json</span>) {</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;<span class="cm-keyword">if</span> (<span class="cm-variable-2">error</span>) <span class="cm-keyword">return</span> <span class="cm-variable">console</span>.<span class="cm-property">warn</span>(<span class="cm-variable-2">error</span>);</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;<span class="cm-variable">asm</span> <span class="cm-operator">=</span> <span class="cm-keyword">new</span> <span class="cm-variable">Assembly</span> (<span class="cm-variable-2">json</span>);</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;<span class="cm-variable">asm</span>.<span class="cm-property">drawPlot</span>(<span class="cm-string">'assembly_stats'</span>);</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  })</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tag cm-bracket">&lt;/</span><span class="cm-tag">script</span><span class="cm-tag cm-bracket">&gt;</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 184px;"></div><div class="CodeMirror-gutters" style="display: none; height: 184px;"></div></div></div></pre></li><li><p><span>Use WebStorm open the porter and visualise the HTML page with &amp;assembly=[name_before_.assembly-stats.json ] modification</span></p></li></ul><p><span>Mt-Example</span></p><p><a href='http://localhost:63342/ninomoriaty/assembly-stats/assembly-stats.html?path=json/&amp;assembly=Danaus_plexippus_v3&amp;view=circle&amp;altView=cumulative&amp;altView=table' target='_blank' class='url'>http://localhost:63342/ninomoriaty/assembly-stats/assembly-stats.html?path=json/&assembly=Danaus_plexippus_v3&view=circle&altView=cumulative&altView=table</a></p><p><span>Mt-Practice</span></p><p><a href='http://localhost:63342/ninomoriaty/assembly-stats/assembly-stats.html?path=json/&amp;assembly=output2&amp;view=circle&amp;altView=cumulative&amp;altView=table' target='_blank' class='url'>http://localhost:63342/ninomoriaty/assembly-stats/assembly-stats.html?path=json/&assembly=output2&view=circle&altView=cumulative&altView=table</a></p><p><a href='http://localhost:63342/ninomoriaty/assembly-stats/assembly-stats.html?path=json/&amp;assembly=Assembly_SRR9198433&amp;view=circle&amp;altView=cumulative&amp;altView=table' target='_blank' class='url'>http://localhost:63342/ninomoriaty/assembly-stats/assembly-stats.html?path=json/&assembly=Assembly_SRR9198433&view=circle&altView=cumulative&altView=table</a></p><h3 id='assemblystats'><span>assembly_stats</span></h3><p><strong><span>Ref</span></strong></p><ul><li></li></ul><p><strong><span>Func</span></strong></p><ul><li><span>Quick Check &#39;Contig Stats&#39; and &#39;Scaffold Stats&#39; for a genome assembly</span></li></ul><p><strong><span>Mt-Install</span></strong></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Tips: assembly_stats not assembly-stats</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">pip install assembly_stats</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 46px;"></div><div class="CodeMirror-gutters" style="display: none; height: 46px;"></div></div></div></pre><h3 id='genomeqc'><span>GenomeQC</span></h3><p><span>Ref</span></p><ul><li><span>GitHub: </span><a href='https://github.com/HuffordLab/GenomeQC' target='_blank' class='url'>https://github.com/HuffordLab/GenomeQC</a></li></ul><h3 id='squat'><span>SQUAT</span></h3><p><strong><span>Ref</span></strong></p><ul><li><span>Article: </span><a href='https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-5445-3' target='_blank' class='url'>https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-5445-3</a></li><li></li></ul><p>&nbsp;</p><h2 id='genome-annotation'><span>Genome Annotation</span></h2><p><font color=orange size=6><span>Ref</span></font></p><ul><li><span>Yandell, M. and Ence, D. (2012) A beginner’s guide to eukaryotic genome annotation. Nature Reviews Genetics 13: 329-342.</span></li><li><span>Stanke, M., Diekhans, M., Baertsch, R. and Haussler, D. (2008) Using native and syntenically mapped cDNA alignments to improve de novo gene finding. Bioinformatics 24: 637-644.</span></li><li><span>Guigó, R., Flicek, P., Abril, J.F., Reymond, A., Lagarde, J., Denoeud, F., Antonarakis, S., Ashburner, M., Bajic, V.B., Birney, E., Castelo, R., Eyras, E., Ucla, C., Gingeras, T.R., Harrow, J., Hubbard, T., Lewis, S.E. and Reese, M.G. (2006) EGASP: the human ENCODE Genome Annotation Assessment Project. Genome Biology 7(Suppl 1): S2.</span></li></ul><hr /><p><font color=orange size=6><span>Pcp</span></font></p><ul><li><span>The optimal parameters for annotation are species specific.</span></li><li></li></ul><hr /><p><font color=orange size=6><span> Cat-General </span></font></p><p><span>Tips: General method usually combine ab initio and homology-based annotation</span></p><h3 id='ncbi-prokaryotic-genome-annotation-process'><span>NCBI Prokaryotic Genome Annotation Process</span></h3><p><span>Ref: </span><a href='https://www.ncbi.nlm.nih.gov/genome/annotation_prok/process' target='_blank' class='url'>https://www.ncbi.nlm.nih.gov/genome/annotation_prok/process</a></p><h3 id='ensembl-genome-annotation-project'><span>EnsEMBL genome annotation project</span></h3><p><span>Ref: </span><a href='http://ensemblgenomes.org/info/data/annotation' target='_blank' class='url'>http://ensemblgenomes.org/info/data/annotation</a></p><h4 id='mt-braker-pipeline'><span>Mt-BRAKER pipeline</span></h4><p><strong><span>Ref</span></strong><span>: Hoff et al. (2019) Whole-Genome Annotation with BRAKER. doi: 10.1007/978-1-4939-9173-0_5</span></p><p><strong><span>Cc</span></strong></p><ul><li><span>Training: an evaluation – perhaps one of a series – to help us set parameter values.</span></li><li><span>Testing: a final evaluation to see how well our parameter values work</span></li><li><span>Sensitivity measures the proportion of positives that are correctly identified.</span></li><li><span>Specificity measures the proportion of negatives that are correctly identified.</span></li><li></li></ul><p><strong><span>Pcp</span></strong></p><ul><li><span>Use GeneMark-ET to produce initial set of protein coding genes for training</span></li><li><span>Use AUGUSTUS to final gene prediction (machine learning)</span></li></ul><p><strong><span>Pc</span></strong></p><ol start='' ><li><p><span>GeneMark-ES/ET </span></p><ol start='' ><li><span>Input: genome.fasta data and extrinsic data (like RNA-seq)</span></li><li><span>output: genemark.gtf</span></li></ol></li><li><p><span>Filter gene</span></p></li><li><p><span>AUGUSTUS Training</span></p></li><li><p><span>AUGUSTUS Prediction</span></p></li><li><p><span>Evaluation with pre-defined known gene</span></p></li><li><p><span>Evaluation with benchmark</span></p><ol start='' ><li><p><strong><span>BUSCO</span></strong><span>: Benchmarking Universal Single-Copy Orthologs</span></p><ol start='' ><li><span>BUSCO genes are expected to be found in 1 copy per genome.</span></li></ol></li><li><p><strong><span>CEGMA</span></strong><span>: Core Eukaryotic Genes Mapping Approach</span></p><ol start='' ><li><span>Each eukaryote is expected to have a set of core genes.</span></li><li><span>Did our annotation correctly predict the CEGMA core?</span></li></ol></li></ol></li><li><p><span>Final output: GFF3 format for the annotation</span></p></li><li><p><span>Prove gene function</span></p><ol start='' ><li><span>Test of molecular function: like knockdowns etc. molecular experiments</span></li><li><span>Test of adaptive function: field selection experiments</span></li></ol></li></ol><p><strong><span>Mt-Improve</span></strong></p><ul><li><span>Use a high quality and high contiguity genome</span></li><li><span>Mask for repeats</span></li><li><span>Consider organism of interest and clade specific features</span></li></ul><p>&nbsp;</p><h3 id='annova'><span>ANNOVA</span></h3><p><span>Cat: General Gene Annotation and other function</span></p><hr /><p><font color=orange size=6><span>Cat- Function annotation</span></font></p><h3 id='gene-ontology'><span>Gene Ontology</span></h3><p><span>Cat: </span></p><h3 id='interproscan'><span>InterProScan</span></h3><p><span>Cat:</span></p><hr /><p><font color=orange size=6><span>Cat-de novo genome annotation</span></font></p><h3 id='genemark-et'><span>GeneMark-ET</span></h3><p><span>Cat: de novo genome annotation</span></p><h3 id='snap'><span>SNAP</span></h3><p><span>Cat: de novo genome annotation</span></p><h3 id='genscan-software'><span>GenScan software</span></h3><p><span>Cat: de novo genome annotation</span></p><hr /><p><font color=orange size=6><span>Cat-Homology-based genome annotation</span></font></p><h3 id='tblastn'><span>TBLASTN</span></h3><p><span>Cat: Homology-based genome annotation</span></p><h3 id='genewise'><span>GeneWise</span></h3><p><span>Cat: Homology-based genome annotation</span></p><h3 id='augustus-software'><span>AUGUSTUS software</span></h3><p><span>Ref</span></p><ul><li><a href='http://bioinf.uni-greifswald.de/augustus/' target='_blank' class='url'>http://bioinf.uni-greifswald.de/augustus/</a></li></ul><p><span>Cat: Homology-based genome annotation</span></p><h3 id='hmmer'><span>HMMER</span></h3><p><span>Ref</span></p><ul><li></li></ul><p><span>Func</span></p><ul><li><span>sequence homologs</span></li><li><span>making sequence alignments</span></li></ul><p>&nbsp;</p><h2 id='conservation-analysis'><span>Conservation Analysis</span></h2><h3 id='for-taxonomic-group'><span>For taxonomic group</span></h3><h3 id='for-genes'><span>For genes</span></h3><p>&nbsp;</p><p>&nbsp;</p><h2 id='emboss'><span>EMBOSS</span></h2><p><span>Tips: European Molecular Biology Open Software Suite</span></p><p><strong><span>Ref</span></strong><span>:</span><a href='http://emboss.sourceforge.net/' target='_blank' class='url'>http://emboss.sourceforge.net/</a></p><p><strong><span>F</span></strong><span>:</span></p><ol start='' ><li><span>provide small and single-purpose bioinformatics tools</span></li><li><span>it provide online </span><a href='http://emboss.bioinformatics.nl/'><span>GUI version</span></a><span> and command line version</span></li></ol><p><strong><span>Mt-Install</span></strong></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">sudo</span> apt install emboss</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 23px;"></div><div class="CodeMirror-gutters" style="display: none; height: 23px;"></div></div></div></pre><p><strong><span>Mt-Plotcon</span></strong><span>:</span></p><ol start='' ><li><span>plot conservation of a sequence  alignment</span></li></ol><p>&nbsp;</p><h2 id='entrez-programming-utilities'><span>Entrez Programming Utilities</span></h2><p><strong><span>Ref</span></strong><span>: </span></p><ol start='' ><li><a href='https://www.ncbi.nlm.nih.gov/books/NBK25501/' target='_blank' class='url'>https://www.ncbi.nlm.nih.gov/books/NBK25501/</a></li><li><a href='https://www.ncbi.nlm.nih.gov/books/NBK179288/' target='_blank' class='url'>https://www.ncbi.nlm.nih.gov/books/NBK179288/</a></li></ol><p><strong><span>Mt-install</span></strong></p><ol start='' ><li><p><span>Current method can be found in </span><a href='https://www.ncbi.nlm.nih.gov/books/NBK179288/' target='_blank' class='url'>https://www.ncbi.nlm.nih.gov/books/NBK179288/</a></p></li><li><p><span>But actually with Ubuntu, you can just install through </span><code>apt</code></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">sudo</span> apt install ncbi-entrez-direct</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 23px;"></div><div class="CodeMirror-gutters" style="display: none; height: 23px;"></div></div></div></pre><p>&nbsp;</p></li><li><p>&nbsp;</p></li></ol><p><strong><span>Mt-How to get all sequences for taxonomic groups with </span><code>wget</code></strong></p><ol start='' ><li><p><span>The </span><code>usehistory=y</code><span> parameter will generate the Web environment (</span><code>&amp;WebEnv</code><span>) and query key (</span><code>&amp;query_key</code><span>) parameters that will specify the location of the retrieved GI numbers (NOTE!) on the Entrez history server. </span></p></li><li><p><span>Let&#39;s specify the retrieval type </span><code>rettype</code><span> as &quot;fasta&quot; and return mode </span><code>retmode</code><span> as &quot;text&quot;. </span></p></li><li><p><span>get the WebEnv and query_key</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Set the taxonomic group and database</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">beast</span><span class="cm-operator">=</span><span class="cm-string">"Cosmoscarta"</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">db</span><span class="cm-operator">=</span><span class="cm-string">"nucleotide"</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># wget the WebEnv and query_key</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">wget</span> <span class="cm-attribute">-qO-</span> <span class="cm-string">"http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=</span><span class="cm-def">$db</span><span class="cm-string">&amp;term=</span><span class="cm-def">$beast</span><span class="cm-string">&amp;usehistory=y"</span> | <span class="cm-builtin">awk</span> <span class="cm-string">'{</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## Is this the third line</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-keyword">if</span><span class="cm-def">(FNR</span><span class="cm-operator">==</span><span class="cm-number">3</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">{</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">split(<span class="cm-def">$0</span>,xml_line_split,<span class="cm-string">"&lt;|&gt;"</span>);</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">query_key</span><span class="cm-operator">=</span>xml_line_split[17];</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">WebEnv</span><span class="cm-operator">=</span>xml_line_split[21];</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">print query_key &gt; <span class="cm-string">"query_key"</span>;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">print WebEnv &gt; <span class="cm-string">"WebEnv"</span>;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-keyword">exit</span>;}}<span class="cm-string">'</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># get the query_key and WebEnv for the sequences download.</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">query_key</span><span class="cm-operator">=</span><span class="cm-quote">$(cat query_key)</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">echo</span> <span class="cm-def">$query_key</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">WebEnv</span><span class="cm-operator">=</span><span class="cm-quote">$(cat WebEnv)</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">echo</span> <span class="cm-def">$WebEnv</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># do the esearch component of the search and get the fasta format sequences</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">wget</span> <span class="cm-attribute">-qO</span> Seqs_I_wanted.fasta <span class="cm-string">"http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=</span><span class="cm-def">$db</span><span class="cm-string">&amp;query_key=</span><span class="cm-def">$query_key</span><span class="cm-string">&amp;WebEnv=</span><span class="cm-def">$WebEnv</span><span class="cm-string">&amp;rettype=fasta&amp;retmode=text"</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 621px;"></div><div class="CodeMirror-gutters" style="display: none; height: 621px;"></div></div></div></pre></li><li><p>&nbsp;</p></li></ol><p><strong><span>Mt-Use the </span><code>edirect</code><span> method to get all sequences for taxonomic groups </span></strong></p><ol start='' ><li><p><code>esearch</code><span> and </span><code>efetch</code></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># get the </span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">esearch <span class="cm-attribute">-db</span> nucleotide <span class="cm-attribute">-query</span> <span class="cm-string">"Cosmoscarta[organism]"</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># get the UID values</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">esearch <span class="cm-attribute">-db</span> nucleotide <span class="cm-attribute">-query</span> <span class="cm-string">"Cosmoscarta[organism]"</span> | efetch <span class="cm-attribute">-db</span> nucleotide <span class="cm-attribute">-format</span> uid &gt; Cosmoscarta.nuc.gis</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># get the accession values</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">esearch <span class="cm-attribute">-db</span> nucleotide <span class="cm-attribute">-query</span> <span class="cm-string">"Cosmoscarta[organism]"</span> | efetch <span class="cm-attribute">-db</span> nucleotide <span class="cm-attribute">-format</span> acc &gt; Cosmoscarta.nuc.acc</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># get the fasta sequences</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">esearch <span class="cm-attribute">-db</span> nucleotide <span class="cm-attribute">-query</span> <span class="cm-string">"Cosmoscarta[organism]"</span> | efetch <span class="cm-attribute">-db</span> nucleotide <span class="cm-attribute">-format</span> fasta &gt; Cosmoscarta.nuc.fa</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># get the record in GenBank format</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">esearch <span class="cm-attribute">-db</span> nucleotide <span class="cm-attribute">-query</span> <span class="cm-number">1802791303</span> | efetch <span class="cm-attribute">-db</span> nucleotide <span class="cm-attribute">-format</span> gb &gt; Cosmoscarta.first.gb</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 437px;"></div><div class="CodeMirror-gutters" style="display: none; height: 437px;"></div></div></div></pre></li><li><p>&nbsp;</p></li><li><p>&nbsp;</p></li></ol><p>&nbsp;</p><h2 id='pullseq'><span>Pullseq</span></h2><p>&nbsp;</p><h2 id='samtools-1'><span>SAMtools</span></h2><h3 id='samtools-2'><span>SAMtools</span></h3><p>&nbsp;</p><p><span>Document: </span></p><p><span>Mt-Common Usage</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Simple stat summary</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">samtools flagstat [file] <span class="cm-comment"># show the SIMPLE smmary statistics on the file</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Convert files Types</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">samtools view <span class="cm-attribute">-b</span> [file.sam] <span class="cm-comment"># SAM &gt; BAM</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">samtools view <span class="cm-attribute">-h</span> [file.bam] <span class="cm-comment"># BAM &gt; SAM</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">samtools fastq [file.bam] <span class="cm-comment"># BAM &gt; FASTQ</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">samtools fasta [file.bam] <span class="cm-comment"># BAM &gt; FASTA</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 207px;"></div><div class="CodeMirror-gutters" style="display: none; height: 207px;"></div></div></div></pre><p><span>Mt-Installation</span></p><ol start='' ><li><a href='http://www.htslib.org/download/' target='_blank' class='url'>http://www.htslib.org/download/</a></li></ol><h3 id='bedtools'><span>BEDtools</span></h3><p><span>Document:</span><a href='https://bedtools.readthedocs.io/en/latest/index.html' target='_blank' class='url'>https://bedtools.readthedocs.io/en/latest/index.html</a></p><ol start='' ><li><p><span>jkl</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.5px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">~/seqtk-master/seqtk trimfq SRR026762.fastq.gz | ~/seqtk-master/seqtk seq <span class="cm-attribute">-a</span> <span class="cm-attribute">-</span> &gt; SRR026762.trimmed.fasta</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 46px;"></div><div class="CodeMirror-gutters" style="display: none; height: 46px;"></div></div></div></pre></li><li><p>&nbsp;</p></li></ol><h3 id='bcftools'><span>BCFtools</span></h3><h3 id='htslib'><span>HTSlib</span></h3><p>&nbsp;</p><h3 id='star'><span>STAR</span></h3><p>&nbsp;</p><h1 id='pipeline-programming-language'><span>Pipeline Programming Language</span></h1><p><span>Tips: This section is not for the detailed pipeline constructions and it provides a index to link those pipelines constructors.</span></p><h3 id='nextflow'><span>Nextflow</span></h3><p><strong><span>Lk</span></strong><span>: </span><a href='https://www.nextflow.io/' target='_blank' class='url'>https://www.nextflow.io/</a></p><p><strong><span>Mt-install</span></strong></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Remote usage</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">java <span class="cm-attribute">--version</span> <span class="cm-comment"># check if the version is sutiable for the usage of nextflow</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">curl</span> <span class="cm-attribute">-s</span> https://get.nextflow.io | <span class="cm-builtin">bash</span> <span class="cm-comment"># download the makefile and use bash to execute the shell script to install and config</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">./nextflow run hello <span class="cm-comment"># test if the installation is complete</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-keyword">export</span> <span class="cm-def">PATH</span><span class="cm-operator">=</span>/home/ninomoriaty/Nextflow:<span class="cm-def">$PATH</span> <span class="cm-comment"># add to .bashrc</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Download by Bioconda</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">conda install <span class="cm-attribute">-c</span> bioconda nextflow</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 207px;"></div><div class="CodeMirror-gutters" style="display: none; height: 207px;"></div></div></div></pre><p>&nbsp;</p><h3 id='snakemake'><span>Snakemake</span></h3><p><strong><span>Lk</span></strong><span>: </span></p><p><strong><span>Mt-install</span></strong></p><p><strong><span>Add-BBMap</span></strong><span> </span></p><p>&nbsp;</p><h1 id='physical-biology-and-structure'><span>Physical Biology and Structure</span></h1><h3 id='pymol'><span>PyMOL</span></h3><p><strong><span>Ref</span></strong><span>: </span></p><p><strong><span>Mt-Installation</span></strong></p><p><strong><span>Mt-Usage</span></strong></p><p>&nbsp;</p><h1 id='phylogeneticevolution-relationship'><span>Phylogenetic/Evolution Relationship</span></h1><h3 id='x-mega--mega-x--mega-11'><span>X Mega / MEGA-X / MEGA 11</span></h3><p><strong><span>Ref</span></strong></p><ul><li><span>Official website: </span><a href='https://www.megasoftware.net/' target='_blank' class='url'>https://www.megasoftware.net/</a></li></ul><p><strong><span>Mt-Installation</span></strong></p><ul><li><p><span>Windows</span></p><ol start='' ><li><span>Find the </span></li></ol></li><li><p><span>Ubuntu</span></p><ul><li><p><span>Download from the web browser</span></p><ul><li><span>Actually it is similar with the process in Windows</span></li><li><span>But the software across all platforms is only the MEGA X.</span></li></ul></li><li><p><span>Download from the conda</span></p></li></ul></li></ul><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><h3 id='beast'><span>BEAST</span></h3><p><span>Tips: Bayesian Evolutionary Analysis Sampling Trees</span></p><p><strong><span>Ref</span></strong><span> </span></p><ul><li><span>Main Page: </span></li><li><span>Community and Tutorial:</span><a href='https://beast.community/first_tutorial' target='_blank' class='url'>https://beast.community/first_tutorial</a></li><li></li></ul><p><strong><span>Mt-Installation</span></strong></p><ul><li><span>Ref: </span><a href='https://beast.community/install_on_unix' target='_blank' class='url'>https://beast.community/install_on_unix</a></li><li></li></ul><p>&nbsp;</p><h1 id='cluster-for-community'><span>Cluster for Community</span></h1><h3 id='netlogo'><span>Netlogo</span></h3><ul><li></li></ul><p>&nbsp;</p><p>&nbsp;</p><table><tbody><tr><td style="background: #3ADDBA !important;"><center><font style="font-size:45px; color:black !important;"><b>
    Database in BioI
    </b></font></center></td></tr></tbody></table><h1 id='2-database-in-bioi'><span>2 Database in BioI</span></h1><h1 id='integrated-database'><span>Integrated Database</span></h1><h3 id='ncbi'><span>NCBI</span></h3><p><span>Ref: </span><a href='http://www.ncbi.nlm.nih.gov/' target='_blank' class='url'>http://www.ncbi.nlm.nih.gov/</a></p><p><span>Tutorial: </span><a href='https://www.ncbi.nlm.nih.gov/books/NBK1058/' target='_blank' class='url'>https://www.ncbi.nlm.nih.gov/books/NBK1058/</a></p><h4 id='ncbi-taxonomy'><span>NCBI taxonomy</span></h4><ol start='' ><li><p><strong><span>Ref</span></strong><span>: </span><a href='http://www.ncbi.nlm.nih.gov/taxonomy' target='_blank' class='url'>http://www.ncbi.nlm.nih.gov/taxonomy</a></p></li><li><p><strong><span>F:</span></strong><span> </span></p><ol start='' ><li><span>Search database in the taxonomic group with the taxon name</span></li><li><span>Get the taxonomy ID and it could be used in other database in NCBI</span></li><li></li></ol></li><li><p>&nbsp;</p></li></ol><h4 id='genbank'><span>GenBank</span></h4><ol start='' ><li><p><strong><span>Pcp-How to use the search syntax in NCBI</span></strong></p><ol start='' ><li><p><span>[] could limit the search range</span></p><ol start='' ><li><span>[Organism:exp] protein sequence</span></li><li><span>[Gene Name] gene name</span></li><li><span>NOT Partial will not include the partial gene/sequence(in another word, it will be complete)</span></li></ol></li><li><p><span>() could prioritise the statement</span></p></li><li><p><span>logic operators</span></p><ol start='' ><li><span>AND</span></li><li><span>OR</span></li><li><span>NOT</span></li></ol></li><li><p>&nbsp;</p></li></ol></li><li><p><strong><span>Cpn-NCBI data information page</span></strong></p><ol start='' ><li><strong><span>Metadata</span></strong><span>: The information about the experiment of the dataset and the sample information of the result</span></li><li><strong><span>Annotation</span></strong><span>: The ontological and functional annotation and index of the sequence</span></li><li><strong><span>Sequences</span></strong><span>: The sequences of the sample.</span></li><li></li></ol></li><li><p><strong><span>F:</span></strong></p><ol start='' ><li><span>Use the taxonomy ID to find query sequences</span></li></ol></li></ol><p>&nbsp;</p><h3 id='european-institute-of-bioinformaticsembl-ebi'><span>European Institute of Bioinformatics(EMBL-EBI)</span></h3><p><strong><span>Ref</span></strong></p><ul><li><a href='https://www.ebi.ac.uk/' target='_blank' class='url'>https://www.ebi.ac.uk/</a></li></ul><p>&nbsp;</p><p>&nbsp;</p><h1 id='nucleotide-database'><span>Nucleotide Database</span></h1><h3 id='european-nucleotide-archiveena'><span>European Nucleotide Archive(ENA)</span></h3><p><strong><span>Ref</span></strong></p><ul><li><a href='https://www.ebi.ac.uk/ena/browser/home' target='_blank' class='url'>https://www.ebi.ac.uk/ena/browser/home</a></li></ul><p><strong><span>Obj2</span></strong></p><ul><li><span>Collect free+public+annotated DNA and RNA sequences</span></li><li></li></ul><p>&nbsp;</p><h3 id='ensemble-biomart'><span>Ensemble BioMart</span></h3><p>&nbsp;</p><h1 id='protein-database'><span>Protein Database</span></h1><h3 id='prosite'><span>PROSITE</span></h3><p>&nbsp;</p><h3 id='uniportswiss-port'><span>UniPort/Swiss Port</span></h3><p><span>Cc</span></p><p>&nbsp;</p><p><span>Ab-Total Database</span></p><ul><li><span>Protein Resource Database</span></li><li><span>Data is manually curated by domain experts</span></li><li></li></ul><p>&nbsp;</p><p><span>Ab-Every part of Annotation Data</span></p><ul><li><span>Protein first/second/quaternary structure</span></li><li><span>Protein domain, sites</span></li><li><span>protein function</span></li><li><span>protein post translation modifications</span></li><li><span>Similarity (Homologene)</span></li><li><span>Potential Sequence variants </span></li><li><span>Disease</span></li><li><span>Gene Ontology Information</span></li></ul><p>&nbsp;</p><h3 id='pdb'><span>PDB</span></h3><p><strong><span>Ref</span></strong><span>:</span><a href='https://www.rcsb.org/' target='_blank' class='url'>https://www.rcsb.org/</a></p><p>&nbsp;</p><h1 id='functional-omics-database'><span>Functional omics Database</span></h1><h3 id='geo'><span>GEO</span></h3><p>&nbsp;</p><h3 id='kegg-pathway'><span>KEGG Pathway</span></h3><p>&nbsp;</p><h1 id='drug-discovery-database'><span>Drug Discovery Database</span></h1><h3 id='drug-bank'><span>Drug Bank</span></h3><p><strong><span>Ref</span></strong><span>: </span><a href='https://go.drugbank.com/' target='_blank' class='url'>https://go.drugbank.com/</a></p><p>&nbsp;</p><h3 id='chembl'><span>ChEMBL</span></h3><p><strong><span>Ref</span></strong><span>: </span><a href='https://www.ebi.ac.uk/chembl/' target='_blank' class='url'>https://www.ebi.ac.uk/chembl/</a></p><p>&nbsp;</p><h3 id='zinc'><span>ZINC</span></h3><p><strong><span>Ref</span></strong><span>: </span><a href='http://zinc15.docking.org' target='_blank' class='url'>http://zinc15.docking.org</a></p><p>&nbsp;</p><p>&nbsp;</p><table><tbody><tr><td style="background: #3ADDBA !important;"><center><font style="font-size:45px; color:black !important;"><b>
    Data Structure in BioI
    </b></font></center></td></tr></tbody></table><h1 id='3-data-structure-in-bioi'><span>3 Data Structure in BioI</span></h1><h1 id='pcp-codes-and-symbols'><span>Pcp-codes and symbols</span></h1><h2 id='iupac-ambiguity-codes'><strong><span>IUPAC Ambiguity Codes</span></strong><span>:</span></h2><p><span>Tips: If needed, you can remove the nucleotide codes which don&#39;t list in this sheet.</span></p><figure><table><thead><tr><th><span>IUPAC Code</span></th><th><span>Meaning</span></th><th><span>Complement</span></th></tr></thead><tbody><tr><td><span>A</span></td><td><span>A</span></td><td><span>T</span></td></tr><tr><td><span>C</span></td><td><span>C</span></td><td><span>G</span></td></tr><tr><td><span>G</span></td><td><span>G</span></td><td><span>C</span></td></tr><tr><td><span>T/U</span></td><td><span>T</span></td><td><span>A</span></td></tr><tr><td><span>M</span></td><td><span>A or C</span></td><td><span>K</span></td></tr><tr><td><span>R</span></td><td><span>A or G</span></td><td><span>Y</span></td></tr><tr><td><span>W</span></td><td><span>A or T</span></td><td><span>W</span></td></tr><tr><td><span>S</span></td><td><span>C or G</span></td><td><span>S</span></td></tr><tr><td><span>Y</span></td><td><span>C or T</span></td><td><span>R</span></td></tr><tr><td><span>K</span></td><td><span>G or T</span></td><td><span>M</span></td></tr><tr><td><span>V</span></td><td><span>A or C or G</span></td><td><span>B</span></td></tr><tr><td><span>H</span></td><td><span>A or C or T</span></td><td><span>D</span></td></tr><tr><td><span>D</span></td><td><span>A or G or T</span></td><td><span>H</span></td></tr><tr><td><span>B</span></td><td><span>C or G or T</span></td><td><span>V</span></td></tr><tr><td><span>N</span></td><td><span>G or A or T or C</span></td><td><span>N</span></td></tr></tbody></table></figure><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><h1 id='fomats'><span>Fomats</span></h1><h2 id='genpept'><span>GenPept</span></h2><p><strong><span>Cpn</span></strong></p><ol start='' ><li><span>Metadata</span></li><li><span>annotation</span></li><li><span>sequence</span></li></ol><p><strong><span>Ab</span></strong></p><ol start='' ><li><span>A complex and strictly defined structure</span></li></ol><p>&nbsp;</p><h2 id='genbankgb'><span>GenBank(gb)</span></h2><p>&nbsp;</p><p><strong><span>Cpn</span></strong></p><ol start='' ><li><span>CDS: the range of coding segment</span></li><li></li></ol><p>&nbsp;</p><p><strong><span>e.g.</span></strong></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">LOCUS &nbsp; &nbsp; &nbsp; AJ223353 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 808 bp &nbsp;  mRNA &nbsp;  linear &nbsp; PRI 07-OCT-2008</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">DEFINITION  Homo sapiens mRNA for histone H2B, clone pJG4-5-15.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">ACCESSION &nbsp; AJ223353</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">VERSION &nbsp; &nbsp; AJ223353.1  GI:3255998</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">KEYWORDS &nbsp;  histone H2B.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">SOURCE &nbsp; &nbsp;  Homo sapiens (human)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  ORGANISM  Homo sapiens</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;  Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;  Mammalia; Eutheria; Euarchontoglires; Primates; Haplorrhini;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;  Catarrhini; Hominidae; Homo.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">REFERENCE &nbsp; 1</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  AUTHORS &nbsp; Lorain,S., Quivy,J.P., Monier-Gavelle,F., Scamps,C., Lecluse,Y.,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;  Almouzni,G. and Lipinski,M.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  TITLE &nbsp; &nbsp; Core histones and HIRIP3, a novel histone-binding protein, directly</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;  interact with WD repeat protein HIRA</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  JOURNAL &nbsp; Mol. Cell. Biol. 18 (9), 5546-5556 (1998)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; PUBMED &nbsp; 9710638</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">REFERENCE &nbsp; 2  (bases 1 to 808)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  AUTHORS &nbsp; Lipinski,M.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  TITLE &nbsp; &nbsp; Direct Submission</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  JOURNAL &nbsp; Submitted (08-JAN-1998) Lipinski M., CNRS URA 1156, Institut</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;  Gustave Roussy, Rue Camille Desmoulins, 94805, FRANCE</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">FEATURES &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; Location/Qualifiers</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; source &nbsp; &nbsp; &nbsp; &nbsp;  1..808</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /organism="Homo sapiens"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /mol_type="mRNA"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /db_xref="taxon:9606"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /clone="pjG4-5-15"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /cell_line="HeLa"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /tissue_type="cervix carcinoma"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /dev_stage="adult"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; gene &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;  1..808</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /gene="HIRIP2"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; CDS &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; 29..409</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /gene="HIRIP2"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /codon_start=1</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /product="Histone H2B"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /protein_id="CAA11277.1"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /db_xref="GI:3255999"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /db_xref="GOA:P58876"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /db_xref="HGNC:HGNC:4747"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /db_xref="InterPro:IPR000558"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /db_xref="InterPro:IPR007125"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /db_xref="InterPro:IPR009072"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /db_xref="UniProtKB/Swiss-Prot:P58876"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /translation="MPEPTKSAPAPKKGSKKAVTKAQKKDGKKRKRSRKESYSVYVYK</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; VLKQVHPDTGISSKAMGIMNSFVNDIFERIAGEASRLAHYNKRSTITSREIQTAVRLL</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; LPGELAKHAVSEGTKAVTKYTSSK"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; regulatory &nbsp; &nbsp;  769..774</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /regulatory_class="polyA_signal_sequence"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /gene="HIRIP2"</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">ORIGIN &nbsp; &nbsp; &nbsp;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp;  1 ggggagtgtg ctaactatta acgctacgat gcctgaacct accaagtctg ctcctgcccc</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; 61 aaagaagggc tccaagaagg cggtgactaa ggctcagaag aaggacggga agaagcgcaa</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;  121 gcgcagccgc aaggagagct attcagtgta tgtgtacaag gtgctgaagc aggtccatcc</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;  181 cgacaccggc atctcttcca aggcaatggg gatcatgaat tccttcgtca acgacatctt</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;  241 cgagcgcatc gcaggcgagg cttcccgcct ggcgcattac aacaagcgct cgaccatcac</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;  301 ctccagggag atccagacgg ccgtgcgcct gctgcttccg ggggagctgg ccaagcacgc</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;  361 cgtgtcggag ggcaccaagg ccgtcaccaa gtacaccagt tccaagtaac tttgccaagg</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;  421 gagagacatg aagacagagg agaaatgaat gcataaaata actgataata tgaatctata</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;  481 catagaactt aggaagtctc atctgcctga aaatgactgt gtggatccca cccaaatcca</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;  541 actcatcctg gtttgctgca cactggttca tcaaaagaag gttaccgagg ggaaggaact</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;  601 aaaggtgttt gcacttcatg ttactttttg agtttataaa cataaaaaca gaatttactt</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;  661 ctgttacaga cctagttact gggaattcat tacttgccat ggactacctt tgctaagaaa</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;  721 agtctgaatg agaagatggc aggacgtctg aaaaaaaaag ttataattaa taaaatctgc</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;  781 ggagaattgt aaaaaaaaaa aaaaaaaa</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">//</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 1564px;"></div><div class="CodeMirror-gutters" style="display: none; height: 1564px;"></div></div></div></pre><p>&nbsp;</p><p>&nbsp;</p><h2 id='fasta-format'><span>FASTA format</span></h2><p><strong><span>Cpn</span></strong></p><ol start='' ><li><span>Brief description/header(with the product information and organism details)</span></li><li><span>Sequnces</span></li></ol><p><strong><span>Ab</span></strong></p><ol start='' ><li><span>Brief and standard for many downstream analysis</span></li></ol><p><strong><span>e.g.</span></strong></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">&gt;gi|3255998|emb|AJ223353.1| Homo sapiens mRNA for histone H2B, clone pJG4-5-15</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">GGGGAGTGTGCTAACTATTAACGCTACGATGCCTGAACCTACCAAGTCTGCTCCTGCCCCAAAGAAGGGC</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">TCCAAGAAGGCGGTGACTAAGGCTCAGAAGAAGGACGGGAAGAAGCGCAAGCGCAGCCGCAAGGAGAGCT</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">ATTCAGTGTATGTGTACAAGGTGCTGAAGCAGGTCCATCCCGACACCGGCATCTCTTCCAAGGCAATGGG</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">GATCATGAATTCCTTCGTCAACGACATCTTCGAGCGCATCGCAGGCGAGGCTTCCCGCCTGGCGCATTAC</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">AACAAGCGCTCGACCATCACCTCCAGGGAGATCCAGACGGCCGTGCGCCTGCTGCTTCCGGGGGAGCTGG</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">CCAAGCACGCCGTGTCGGAGGGCACCAAGGCCGTCACCAAGTACACCAGTTCCAAGTAACTTTGCCAAGG</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">GAGAGACATGAAGACAGAGGAGAAATGAATGCATAAAATAACTGATAATATGAATCTATACATAGAACTT</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">AGGAAGTCTCATCTGCCTGAAAATGACTGTGTGGATCCCACCCAAATCCAACTCATCCTGGTTTGCTGCA</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">CACTGGTTCATCAAAAGAAGGTTACCGAGGGGAAGGAACTAAAGGTGTTTGCACTTCATGTTACTTTTTG</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">AGTTTATAAACATAAAAACAGAATTTACTTCTGTTACAGACCTAGTTACTGGGAATTCATTACTTGCCAT</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">GGACTACCTTTGCTAAGAAAAGTCTGAATGAGAAGATGGCAGGACGTCTGAAAAAAAAAGTTATAATTAA</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">TAAAATCTGCGGAGAATTGTAAAAAAAAAAAAAAAAAA</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 299px;"></div><div class="CodeMirror-gutters" style="display: none; height: 299px;"></div></div></div></pre><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><h2 id='fastq-format'><span>FASTQ format</span></h2><p><strong><span>Soc</span></strong><span>:</span><a href='https://support.illumina.com/bulletins/2016/04/fastq-files-explained.html'><span>Illumina&#39;s fastq-files-explained.html</span></a></p><p><strong><span>e.g.</span></strong></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">@K00114:84:H2N7KBBXX:8:1101:6105:1457 1:N:0</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">ATTTCCTTTTCGTGGATACGACANAGGATGATGCAAGCAGCGAGTAAACGATTTGATGAGGCTTAGCGCATCATGAATAGCACAGATGATGTGGAGAAAG</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">+</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">AAFFFKFFKKKKKKKKKKKFKKK#KFKFFFAKKKKF7,AFAKAAKKKKKK,FAFFFKA7FFKKFK,,7AFFKAKKKKKF,7FKK7AFFF,A,,,7AFKF7</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">@K00114:84:H2N7KBBXX:8:1101:6856:1457 1:N:0</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">TGAAGAAATAAAACAAGTAACAGNGACACCAACTCAGCTATCAGCCGCTTCCGTGATTAGACGCAGCCGTTCAACGGCCATTGCGAGCAAGGCAAATGTA</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">+</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">AAFFFKKKKKKKKKKKKKKKKKK#KKKKKKKKKKKKKKKKKFKFFAFFKKKKKKKKKKKFKKFFAF,(7AFAFFKFFKFFKKKKKK7FKKKKKKKKKA,A</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">...</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 299px;"></div><div class="CodeMirror-gutters" style="display: none; height: 299px;"></div></div></div></pre><p>&nbsp;</p><h2 id='blast-output'><span>BLAST output</span></h2><p><strong><span>e.g.</span></strong></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"># BLASTX 2.2.28 + </span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"># Query: gi|322830704:1426-2962 Boreus elegans mitochondrion, complete genome</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"># Database: nem</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"># Fields: query id, subject id, % identity, alignment length, mismatches, gap opens, q. start, q. end, s. start, s. end, evalue, bit score</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"># 405 hits found</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gi|322830704:1426-2962  gi|225622197|ref|YP_002725710.1| &nbsp; &nbsp; &nbsp;  61.71 &nbsp; 444 &nbsp; &nbsp; 169 &nbsp; &nbsp; 1 &nbsp; &nbsp; &nbsp; 4 &nbsp; &nbsp; &nbsp; 1335 &nbsp;  12 &nbsp; &nbsp;  454 &nbsp; &nbsp; 4e-145 &nbsp; 432</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gi|322830704:1426-2962  gi|288903410|ref|YP_003434131.1| &nbsp; &nbsp; &nbsp;  61.99 &nbsp; 442 &nbsp; &nbsp; 167 &nbsp; &nbsp; 1 &nbsp; &nbsp; &nbsp; 10 &nbsp; &nbsp;  1335 &nbsp;  14 &nbsp; &nbsp;  454 &nbsp; &nbsp; 2e-144 &nbsp; 430</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gi|322830704:1426-2962  gi|326536058|ref|YP_004300493.1| &nbsp; &nbsp; &nbsp;  62.44 &nbsp; 442 &nbsp; &nbsp; 165 &nbsp; &nbsp; 1 &nbsp; &nbsp; &nbsp; 10 &nbsp; &nbsp;  1335 &nbsp;  14 &nbsp; &nbsp;  454 &nbsp; &nbsp; 2e-144 &nbsp; 430</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gi|322830704:1426-2962  gi|188011119|ref|YP_001905892.1| &nbsp; &nbsp; &nbsp;  61.31 &nbsp; 442 &nbsp; &nbsp; 170 &nbsp; &nbsp; 1 &nbsp; &nbsp; &nbsp; 10 &nbsp; &nbsp;  1335 &nbsp;  14 &nbsp; &nbsp;  454 &nbsp; &nbsp; 4e-144 &nbsp; 429</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gi|322830704:1426-2962  gi|225622184|ref|YP_002725698.1| &nbsp; &nbsp; &nbsp;  61.49 &nbsp; 444 &nbsp; &nbsp; 170 &nbsp; &nbsp; 1 &nbsp; &nbsp; &nbsp; 4 &nbsp; &nbsp; &nbsp; 1335 &nbsp;  12 &nbsp; &nbsp;  454 &nbsp; &nbsp; 6e-144 &nbsp; 429</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gi|322830704:1426-2962  gi|171260186|ref|YP_001795390.1| &nbsp; &nbsp; &nbsp;  61.54 &nbsp; 442 &nbsp; &nbsp; 169 &nbsp; &nbsp; 1 &nbsp; &nbsp; &nbsp; 10 &nbsp; &nbsp;  1335 &nbsp;  15 &nbsp; &nbsp;  455 &nbsp; &nbsp; 6e-144 &nbsp; 429</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gi|322830704:1426-2962  gi|288903312|ref|YP_003434040.1| &nbsp; &nbsp; &nbsp;  61.99 &nbsp; 442 &nbsp; &nbsp; 167 &nbsp; &nbsp; 1 &nbsp; &nbsp; &nbsp; 10 &nbsp; &nbsp;  1335 &nbsp;  14 &nbsp; &nbsp;  454 &nbsp; &nbsp; 6e-144 &nbsp; 429</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gi|322830704:1426-2962  gi|49146527|ref|YP_026087.1| &nbsp;  61.54 &nbsp; 442 &nbsp; &nbsp; 169 &nbsp; &nbsp; 1 &nbsp; &nbsp; &nbsp; 10 &nbsp; &nbsp;  1335 &nbsp;  14 &nbsp; &nbsp;  454 &nbsp; &nbsp; 9e-144 &nbsp; 428</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gi|322830704:1426-2962  gi|296315379|emb|CAQ56159.1| &nbsp;  61.54 &nbsp; 442 &nbsp; &nbsp; 169 &nbsp; &nbsp; 1 &nbsp; &nbsp; &nbsp; 10 &nbsp; &nbsp;  1335 &nbsp;  14 &nbsp; &nbsp;  454 &nbsp; &nbsp; 1e-143 &nbsp; 428</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gi|322830704:1426-2962  gi|160694986|ref|YP_001552232.1|COX1_21467 &nbsp; &nbsp;  61.99 &nbsp; 442 &nbsp; &nbsp; 167 &nbsp; &nbsp; 1 &nbsp; &nbsp; &nbsp; 10 &nbsp; &nbsp;  1335 &nbsp;  14 &nbsp; &nbsp;  454 &nbsp; &nbsp; 1e-143 &nbsp; 428</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">...</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 644px;"></div><div class="CodeMirror-gutters" style="display: none; height: 644px;"></div></div></div></pre><p>&nbsp;</p><h2 id='bed-format'><span>BED format</span></h2><p>&nbsp;</p><p>&nbsp;</p><h2 id='sam-format'><span>SAM format</span></h2><p><span>Soc:</span></p><ol start='' ><li><span>SAMtools</span></li><li><a href='https://samtools.github.io/hts-specs/SAMv1.pdf' target='_blank' class='url'>https://samtools.github.io/hts-specs/SAMv1.pdf</a></li></ol><p><span>Soc-Produce SAM file</span></p><ol start='' ><li><span>bowtie, bwa, STAR, etc. will output in SAM</span></li><li></li></ol><p><span>Soc-Need SAM file</span></p><p><span>Cpn</span></p><ol start='' ><li><span>header section (@): contigs in the reference genome, mapped software</span></li><li><span>data section: sequences listed in lines</span></li></ol><p><span>F:</span></p><ol start='' ><li><span>capture sequences and quality information, as found in FASTQ files</span></li><li><span>capture how the reads align to a reference genome</span></li></ol><p><span>e.g.</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">@HD VN:1.5 SO:coordinate</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">@SQ SN:ref LN:45</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">r001 &nbsp; 99 ref  7 30 8M2I4M1D3M = 37  39 TTAGATAAAGGATACTG *</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">r002 &nbsp;  0 ref  9 30 3S6M1P1I4M *  0 &nbsp; 0 AAAAGATAAGGATA &nbsp;  *</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">r003 &nbsp;  0 ref  9 30 5S6M &nbsp; &nbsp; &nbsp; *  0 &nbsp; 0 GCCTAAGCTAA &nbsp; &nbsp; &nbsp; * SA:Z:ref,29,-,6H5M,17,0;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">r004 &nbsp;  0 ref 16 30 6M14N5M &nbsp;  *  0 &nbsp; 0 ATAGCTTCAGC &nbsp; &nbsp; &nbsp; *</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">r003 2064 ref 29 17 6H5M &nbsp; &nbsp; &nbsp; *  0 &nbsp; 0 TAGGC &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; * SA:Z:ref,9, + ,5S6M,30,1;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">r001  147 ref 37 30 9M &nbsp; &nbsp; &nbsp; &nbsp; =  7 -39 CAGCGGCAT &nbsp; &nbsp; &nbsp; &nbsp; * NM:i:1</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">...</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 207px;"></div><div class="CodeMirror-gutters" style="display: none; height: 207px;"></div></div></div></pre><p>&nbsp;</p><h2 id='bam-format'><span>BAM format</span></h2><p><span>Def: Native compressed format of SAM</span></p><p><span>F</span></p><ol start='' ><li><span>perform more efficient operations on the BAM files than SAM files</span></li></ol><p>&nbsp;</p><h2 id='vcf-format'><span>VCF format</span></h2><p><strong><span>Def</span></strong><span>: Variant Call Format is a text file format (often stored in a compressed manner as .vcf.gz) to store variants</span></p><p><strong><span>Soc</span></strong><span>: </span><a href='https://github.com/samtools/hts-specs' target='_blank' class='url'>https://github.com/samtools/hts-specs</a></p><p><strong><span>Cpn</span></strong><span>:</span></p><ol start='' ><li><p><span>meta-information lines which introduce the sample information</span></p></li><li><ol start='' ><li><span>Ab: preceded with ##</span></li></ol></li><li><p><span>a header line + 3 = one data section</span></p></li><li><ol start='' ><li><p><span>Cpn:</span></p></li><li><ol start='' ><li><span>CHROM: chromosome</span></li><li><span>POS: position</span></li><li><span>ID: variant identifier</span></li><li><span>REF: reference base(s)</span></li><li><span>ALT: alternate base(s)</span></li><li><span>QUAL: quality (not used in our file)</span></li><li><span>FILTER: filter status (not us ed in our file)</span></li><li><span>INFO: additional information – in our file this field contain the variant type (SNV, insertion, or deletion)</span></li></ol></li><li><p><span>Pcp: </span></p></li><li><ol start='' ><li><span>insertion and deletion&#39;s REF and ALT will start with the base that is present before the insertion or deletion event</span></li></ol></li></ol></li><li><p><span>data lines + 2 = one data section </span></p></li><li><p><span>genotype information</span></p></li></ol><p><strong><span>e.g.</span></strong></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">##fileformat=VCFv4.0</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">##fileDate=20090805</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">##source=myImputationProgramV3.1</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">##reference=1000GenomesPilot-NCBI36</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">##phasing=partial</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">##INFO=&lt;ID=NS,Number=1,Type=Integer,Description="Number of Samples With Data"&gt;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">##INFO=&lt;ID=DP,Number=1,Type=Integer,Description="Total Depth"&gt;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">##INFO=&lt;ID=AF,Number=.,Type=Float,Description="Allele Frequency"&gt;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">##INFO=&lt;ID=AA,Number=1,Type=String,Description="Ancestral Allele"&gt;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">##INFO=&lt;ID=DB,Number=0,Type=Flag,Description="dbSNP membership, build 129"&gt;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">##INFO=&lt;ID=H2,Number=0,Type=Flag,Description="HapMap2 membership"&gt;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">##FILTER=&lt;ID=q10,Description="Quality below 10"&gt;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">##FILTER=&lt;ID=s50,Description="Less than 50% of samples have data"&gt;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">##FORMAT=&lt;ID=GT,Number=1,Type=String,Description="Genotype"&gt;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">##FORMAT=&lt;ID=GQ,Number=1,Type=Integer,Description="Genotype Quality"&gt;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">##FORMAT=&lt;ID=DP,Number=1,Type=Integer,Description="Read Depth"&gt;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">##FORMAT=&lt;ID=HQ,Number=2,Type=Integer,Description="Haplotype Quality"&gt;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">#CHROM POS &nbsp; &nbsp; ID &nbsp; &nbsp; &nbsp;  REF ALT &nbsp;  QUAL FILTER INFO &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;  FORMAT &nbsp; &nbsp;  NA00001 &nbsp; &nbsp; &nbsp;  NA00002 &nbsp; &nbsp; &nbsp;  NA00003</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">20 &nbsp; &nbsp; 14370 &nbsp; rs6054257 G &nbsp; &nbsp;  A &nbsp; &nbsp; &nbsp; 29 &nbsp; PASS &nbsp; NS=3;DP=14;AF=0.5;DB;H2 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; GT:GQ:DP:HQ 0|0:48:1:51,51 1|0:48:8:51,51 1/1:43:5:.,.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">20 &nbsp; &nbsp; 17330 &nbsp; . &nbsp; &nbsp; &nbsp; &nbsp; T &nbsp; &nbsp;  A &nbsp; &nbsp; &nbsp; 3 &nbsp;  q10 &nbsp;  NS=3;DP=11;AF=0.017 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; GT:GQ:DP:HQ 0|0:49:3:58,50 0|1:3:5:65,3 &nbsp; 0/0:41:3</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">20 &nbsp; &nbsp; 1110696 rs6040355 A &nbsp; &nbsp;  G,T &nbsp; &nbsp; 67 &nbsp; PASS &nbsp; NS=2;DP=10;AF=0.333,0.667;AA=T;DB GT:GQ:DP:HQ 1|2:21:6:23,27 2|1:2:0:18,2 &nbsp; 2/2:35:4</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">20 &nbsp; &nbsp; 1230237 . &nbsp; &nbsp; &nbsp; &nbsp; T &nbsp; &nbsp;  . &nbsp; &nbsp; &nbsp; 47 &nbsp; PASS &nbsp; NS=3;DP=13;AA=T &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; GT:GQ:DP:HQ 0|0:54:7:56,60 0|0:48:4:51,51 0/0:61:2</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">20 &nbsp; &nbsp; 1234567 microsat1 GTCT &nbsp; G,GTACT 50 &nbsp; PASS &nbsp; NS=3;DP=9;AA=G &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;  GT:GQ:DP &nbsp;  0/1:35:4 &nbsp; &nbsp; &nbsp; 0/2:17:2 &nbsp; &nbsp; &nbsp; 1/1:40:3</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">...</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 713px;"></div><div class="CodeMirror-gutters" style="display: none; height: 713px;"></div></div></div></pre><p>&nbsp;</p><p>&nbsp;</p><h2 id='ccf-format'><span>CCF format</span></h2><h2 id='maf-format'><span>MAF format</span></h2><p><span> </span></p><p>&nbsp;</p><table><tbody><tr><td style="background: #3ADDBA !important;"><center><font style="font-size:45px; color:black !important;"><b>
    High-Performance Computation
    </b></font></center></td></tr></tbody></table><h1 id='3-high-performance-computation'><span>3 High-Performance Computation</span></h1><h1 id='super-computer'><span>Super Computer</span></h1><h2 id='tianhe2'><span>Tianhe2</span></h2><p><span>Source: Guangzhou SYSU</span></p><h2 id='cirsus'><span>Cirsus</span></h2><p>&nbsp;</p><h1 id='cluster-server'><span>Cluster Server</span></h1><h2 id='eddie'><span>Eddie</span></h2><p><strong><span>Ref</span></strong><span>: </span><a href='https://www.wiki.ed.ac.uk/display/ResearchServices/Eddie' target='_blank' class='url'>https://www.wiki.ed.ac.uk/display/ResearchServices/Eddie</a></p><p><strong><span>Source</span></strong><span>: the University of Edinburgh</span></p><p><strong><span>Def</span></strong><span>: Eddie Mark 3 is the third iteration of the University&#39;s </span><u><span>compute cluster</span></u><span> and is available to all University of Edinburgh researchers. It consists of over 7000 cores with up to 3 TB of memory.</span></p><p><strong><span>Cpn</span></strong><span>:</span></p><ol start='' ><li><p><strong><span>node</span></strong><span>:</span></p><ol start='' ><li><p><span>Cat-login nodes and clusters</span></p><ol start='' ><li><span>login nodes: The nodes for interact with the Eddie server and access to the cluster by queuing system</span></li><li><span>Clusters/Working nodes: The nodes for executing the tasks submitted by users</span></li><li><span>Grid Engine/Queuing System: The system for arrange the working sequences for different tasks.</span></li></ol></li><li><p><span>Cat-level of cores and memory</span></p><ol start='' ><li><p><span>Soc-Service: </span><a href='https://www.wiki.ed.ac.uk/display/ResearchServices/Eddie' target='_blank' class='url'>https://www.wiki.ed.ac.uk/display/ResearchServices/Eddie</a></p></li><li><p><span>Soc-Storage: </span><a href='https://www.wiki.ed.ac.uk/display/ResearchServices/Storage' target='_blank' class='url'>https://www.wiki.ed.ac.uk/display/ResearchServices/Storage</a></p></li><li><p><span>Lk: you can find the &quot;The list of all nodes available as well as their specifications&quot; in the Eddie Wiznote in BPSM folder</span></p></li><li><p><span>standard</span></p><ol start='' ><li><span>16 core per nodes</span></li><li><span>64 GB</span></li></ol></li><li><p><span>There will be a list of the nodes that could be used in the official website </span></p></li></ol></li></ol></li><li><p><strong><span>Cores</span></strong><span>: The processing units(Threads) of each of the computers in the cluster</span></p><ol start='' ><li><span>Available memory for each core is the average of the whole memory</span></li></ol></li><li><p><strong><span>Memory:</span></strong><span> RAM/memory</span></p></li></ol><p><strong><span>Ab</span></strong><span>:</span></p><ol start='' ><li><span>Parallelisation: divide the tasks into a number of cores.</span></li><li><span>Different research groups have different Eddie directory and different DataStore space.</span></li></ol><p><strong><span>Mt-The workflow of Eddie</span></strong><span>:</span></p><p><span>P:</span></p><ol start='' ><li><span>You should be within the university network. VPN and other methods are also acceptable.</span></li><li></li></ol><p><span>Pc:</span></p><ol start='' ><li><p><strong><span>Login in the Eddie</span></strong></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">ssh</span> <span class="cm-attribute">-X</span> s2059232@eddie.ecdf.ed.ac.uk</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">ssh</span> <span class="cm-attribute">-X</span> s2059232@129.215.120.98</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 46px;"></div><div class="CodeMirror-gutters" style="display: none; height: 46px;"></div></div></div></pre><p><span>F:</span></p><ol start='' ><li><span>The login space is the home directory for your account/UUN and the documents here will be backed up (up to 2GB space)</span></li><li><span>The scratch space use up to 20-50TB  and it is not backed up, which will be clean after one month.</span></li><li></li></ol></li><li><p><strong><span>Copy the dataset to the scratch space/Move files from MSc server</span></strong></p><ol start='' ><li><p><span>Lk: DataStore: university&#39;s file system for storing research data</span><a href='https://www.wiki.ed.ac.uk/display/ResearchServices/Data+Staging' target='_blank' class='url'>https://www.wiki.ed.ac.uk/display/ResearchServices/Data+Staging</a></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">scp [options] [dir/file] [destination_dir]</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## e.g.</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># copy to the home dir</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">scp ./environment.yml s2059232@eddie.ecdf.ed.ac.uk:/home/s2059232</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># copy to the scratch space</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">scp ./environment.yml s2059232@eddie.ecdf.ed.ac.uk:/exports/eddie/scratch/s2059232/</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># copy to the home dir</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">scp s2059232@eddie.ecdf.ed.ac.uk:/home/s2059232/environment.yml ./</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># copy from the scratch space</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">scp s2059232@eddie.ecdf.ed.ac.uk:/exports/eddie/scratch/s2059232/file ./</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># move the folder and files in a dir to Eddie</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">scp <span class="cm-attribute">-r</span> ./environment.yml s2059232@eddie.ecdf.ed.ac.uk:/home/s2059232/</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># options</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">-r</span> <span class="cm-comment"># recursive</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Tips: After enter the interactive node as ii part, you can use the Unix command to move files from Eddie to DataStore.</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 437px;"></div><div class="CodeMirror-gutters" style="display: none; height: 437px;"></div></div></div></pre></li><li><p><span>Att: BEFORE you log in to eddie (or from another terminal), you will need to scp a file from the bioinformatics server to eddie. </span></p></li><li><p><span>Att: Other Edinburgh University storage solutions can be used to store files and move the files to Eddie with a staging node.</span></p></li></ol></li><li><p><strong><span>Start an interactive session and login in the login nodes:</span></strong></p><p><span>Tips: Lk about how to settle the interactive sesssion:</span><a href='https://www.wiki.ed.ac.uk/display/ResearchServices/Interactive+Sessions' target='_blank' class='url'>https://www.wiki.ed.ac.uk/display/ResearchServices/Interactive+Sessions</a><span> </span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># interactive sessions within the Eddie</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">qlogin</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># indicate the core and memory th</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">qlogin <span class="cm-attribute">-pe</span> sharedmem &lt; NUMBER OF CORES &gt; <span class="cm-attribute">-l</span> <span class="cm-def">h_vmem</span><span class="cm-operator">=</span>&lt; MEMORY PER CORE &gt;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Log into a staging node</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">qlogin <span class="cm-attribute">-q</span> staging</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 138px;"></div><div class="CodeMirror-gutters" style="display: none; height: 138px;"></div></div></div></pre><ol start='' ><li><p><span>Load modules</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># usage</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">module load &lt;modules needed to use <span class="cm-keyword">in</span> Eddie&gt;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">module available <span class="cm-comment"># check available modules</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">## e.g.</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">module load htop</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">module load python</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 161px;"></div><div class="CodeMirror-gutters" style="display: none; height: 161px;"></div></div></div></pre><p><span>F: However, this function could not be tested in my computer thus you may need to try it when the section is needed.</span></p></li></ol></li><li><p><strong><span>Write and configure a shell script</span></strong></p><ol start='' ><li><p><span>Lk: </span><a href='https://www.wiki.ed.ac.uk/display/ResearchServices/Job+Submission' target='_blank' class='url'>https://www.wiki.ed.ac.uk/display/ResearchServices/Job+Submission</a></p></li><li><p><span>Mt: nano or nedit</span></p></li><li><p><span>Mt-Add task description for queuing system</span></p><p><span>Tips: This description must be added to the top of the script and start with </span><code>#$</code><span>. But </span><strong><span>after the shebang line.</span></strong></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="latex" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="latex"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">#!/bin/bash <span class="cm-comment">%shebang line should be in the first line as well</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">% The Grid Engine Instrcution Part</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">#<span class="cm-keyword">$</span> -<span class="cm-variable-2">cwd</span> <span class="cm-comment">% Tell Eddie to run the code in the current working directroy</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">#<span class="cm-keyword">$</span> -t 1-100 <span class="cm-comment">% Grid Engine parameter -t which will give each of our jobs a number (1-100), called the SGE_TASK_ID (for files with numeric suffixes or prefixes and the $SGE_TASK_ID can be called in the shell script. Also ues to parallelise these tasks and consider them as a collection of tasks.)</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">#<span class="cm-keyword">$</span> -<span class="cm-variable-2">N</span> <span class="cm-variable-2">Hello</span> <span class="cm-comment">% Give a name for the task and Eddie will also give the number code for the task name</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">#<span class="cm-keyword">$</span> -l h_rt=01:01:10 <span class="cm-comment">% Specify the time for the job. h_rt means the hoour time: 1 hr and 1 min and 1 sec.</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">#<span class="cm-keyword">$</span> -<span class="cm-variable-2">l</span> <span class="cm-variable-2">h</span><span class="cm-tag">_</span><span class="cm-variable-2">vmem</span>=<span class="cm-number">1</span><span class="cm-variable-2">G</span> <span class="cm-comment">% Specify the total memory(seems that it is not the memory needed for each core?)</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">% The main program</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">####### your commands or original shell script ######</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">% This output part should be placed after the main project.</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">#<span class="cm-keyword">$</span> -<span class="cm-variable-2">o</span> <span class="cm-variable-2">hello</span><span class="cm-error">.</span><span class="cm-variable-2">o</span> <span class="cm-comment">% Specify the output directory and output file</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">#<span class="cm-keyword">$</span> -e hello.e <span class="cm-comment">% Specify the error directory and error report file</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 506px;"></div><div class="CodeMirror-gutters" style="display: none; height: 506px;"></div></div></div></pre><p><span>Tips: To make the command after </span><code>#$</code><span> visible, we use latex as the highlight marker of code section. However, these code sections should be considered as part of the shell script.</span></p></li><li><p><span>Tips: but actually you can use the method in the i part to move scripts into Eddie and execute them as Unix command line. </span></p></li><li><p><span>Add the task into the Eddie clusters and queuing system</span></p></li></ol></li><li><p><strong><span>Submit and complete the analysis</span></strong></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 20.75px; left: 14.25px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Check the manual of the qsub</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">man qsub</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Submit the shell script and wait for the execution</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">qsub &lt;your_script_or_other_execuatble_files&gt;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Display the status of the Eddie</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">qstat <span class="cm-comment"># The state column tells us that there are six running (r state) and the rest are qwstate (queued and waiting). If no output, it means that there is no works for Eddie to do in this accout.</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Kill a job you subimitted to Eddie by the task number or the job name</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">qdel &lt;job_number_or_name&gt;</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># After the job has been finised, get useful info from the Eddie report</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">qacct <span class="cm-attribute">-j</span> &lt;job_number_or_name&gt;</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 276px;"></div><div class="CodeMirror-gutters" style="display: none; height: 276px;"></div></div></div></pre><p>&nbsp;</p><ol start='' ><li></li></ol></li><li><p><span>Copy the results elsewhere</span></p></li><li><p><span>Ask to work inside a node interactively</span></p></li><li><p><span>Check the processes</span></p></li><li><p>&nbsp;</p></li></ol><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p></div></div>
</body>
</html>